---
title: Assignments 2 and 3
author: Erick Rios
date: 2023-02-06
date-modified: 2023-02-14
categories: [code, analysis]
image: hungryyeast.png
description: Insert witty description
code-overflow: wrap
code-tools: true
---

# Goal of Assignment 2

In the quest to ensure data reproducibility, my goals are to:

1.  Collect data of interest.
2.  Import my data of interest into Rstudio.
3.  Describe the data.
4.  If time permits, host the data on a repository; probably github.

```{r}
#| output: false
library(tidyverse)
```

# Importation of 'Proteomic' Data

```{r}
Proteomic.Data <- read.csv("MyData/2022-03-21_trim-stacked.csv")
head(Proteomic.Data)

```

# Data Collection

The 'proteomic' data are the processed results of Data Independent Acquisition Mass Spectrometry experiments that quantified the relative protein abundances in cell lysates of mutant yeast compared to those of wild-type yeast. After processing the signal intensities of the 2 spectra (mutant vs wildtype), the calculated measurement is presented as Log base 2 Fold Change. Resources and information regarding proteins detected in our experiments are available at [Saccharomyces Genome Database](https://www.yeastgenome.org).

# Summary of Proteomic Data

An advantage of the glimpse command is a neater output of the dataframe dimensions, column names, column class type, and 'sample' view of the actual data in the columns.

```{r}
glimpse(Proteomic.Data)

```

# Data Dictionary

What are these columns and what do the values contained in them even represent? What on earth is a 'Gravy.Score'? Fortunately, I have created a legend containing variable (attribute) descriptions.

```{r}
Attributes <- read.csv("MyData/Column_Descriptions.csv")

knitr::kable(Attributes)
```

# 1st Action-Target pair: Search:Explore - Extremes

Now that we have an understanding of the data we are working with. It is time to start using this data to answer a variety of questions related to my research and field. One of the most well known things in my field, is that perturbing Hsp90 function could affect the steady state levels of gene products (proteins) directly and indirectly. Therefore, first question is:

## How many, if any, proteins had a significant change in their steady-state levels in at least one mutant condition?

```{r}
Gene <- unique(Proteomic.Data$Gene[abs(Proteomic.Data$logFC) >= 1.5 &
                                     Proteomic.Data$P.Value <= 0.055])
```

```{r}
#| code-fold: true
Hits <- paste(length(Gene), "out of", length(unique(Proteomic.Data$Gene)), "proteins were significantly affected in at least one mutant condition; Here we consider any protein whose steady state level changed by at least 1.5 (|Log2FC|)" )

knitr::kable(Hits)
```

Considering we found these genes of interest using the attributes 'logFC' and 'P.Value'; lets use these attributes to visualize our findings.

```{r}
#| code-fold: true
x <- Proteomic.Data$logFC
y <- -log10(Proteomic.Data$P.Value)

group <- rep(1, nrow(Proteomic.Data))               # Create group variable
group[abs(x) >= 1.5 & y >= -log10(0.055)] <- 2
group[abs(x) >= 1.5 & y <= -log10(0.055)] <- 3

plot(x, y, 
     col = group, 
     pch = 20,
     ylim = c(0,25),
     xlim = c(-6.5, 6.5),
     xlab = expression("Log"[2]*"Fold Change"),
     ylab = expression("-Log"[10]*"(P.Value)")
)
abline(v = c(1.5, -1.5), col = "red", lty = 2)
abline(h = -log10(0.055), col = "red", lty = 2)
text(x = 1.45, y = 24, labels = "Decreased steady state levels \n p value < 0.055",
     pos = 4, offset = .5)
text(x = -1.45, y = 24, labels = "Increased steady state levels \n p value < 0.055",
     pos = 2, offset = .5)
text(x = -4, y = -log10(0.055), labels = "p value > 0.055", pos = 1, offset = .5)
text(x = 4, y = -log10(0.055), labels = "p value > 0.055", pos = 1, offset = .5)
```

With this 'volcano' plot we visualized what was done prior when we searched for any proteins that met our criteria for significance. Those that met our criteria were marked pink and those that did not meet our criteria were marked black or light green. To better visualize the 'barriers' for significance, red-dashed lines were added. Proteins with positive logFC values had decreased levels and those with negative logFC values had increased levels. Although not relevant for future analysis, we can spot proteins with large changes in steady state levels, but unfortunately did not meet our threshold for statistical significance (light green) and this is was due to too much variance between the 3 biological replicates in the measurement of that protein.
