<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.269">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Erick Rios">
<meta name="dcterms.date" content="2023-02-06">

<title>My Data Blog - Post With Code</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../../styles.css">
</head>

<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">My Data Blog</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../about.html">
 <span class="menu-text">Erick Rios</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/"><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com"><i class="bi bi-twitter" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <div class="quarto-title-block"><div><h1 class="title">Post With Code</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
                                <div class="quarto-categories">
                <div class="quarto-category">news</div>
                <div class="quarto-category">code</div>
                <div class="quarto-category">analysis</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Erick Rios </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">February 6, 2023</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<div class="callout-note callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>Nice work in general. Some minor editing of your section titles and some of the text. Added a hyperlink to the relevant online database. Youâ€™ve done a good job adding explanation to the code. A new person could read this and understand (and use) the data.</p>
</div>
</div>
<section id="goal-of-assignment-2" class="level1">
<h1>Goal of Assignment 2</h1>
<p>In the quest to ensure data reproducibility, my goals are to:</p>
<ol type="1">
<li>Collect data of interest.</li>
<li>Import my data of interest into Rstudio.</li>
<li>Describe the data.</li>
<li>If time permits, host the data on a repository; probably github.</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="importation-of-proteomic-data" class="level1">
<h1>Importation of â€˜Proteomicâ€™ Data</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>Proteomic.Data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"MyData/2022-03-21_trim-stacked.csv"</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(Proteomic.Data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  Systemic.name         contrast UniprotID Gene      logFC    P.Value Length
1         Q0140 WTHsc82_vs_A583T    P02381 VAR1 -0.3271929 0.19173919    398
2         Q0140 WTHsc82_vs_K102E    P02381 VAR1  0.3134806 0.21029408    398
3         Q0140 WTHsc82_vs_W296A    P02381 VAR1  0.1298284 0.59829379    398
4         Q0140 WTHsc82_vs_G309S    P02381 VAR1  0.2124945 0.39108154    398
5         Q0140 WTHsc82_vs_G424D    P02381 VAR1 -0.2209043 0.37293750    398
6         Q0140 WTHsc82_vs_S481Y    P02381 VAR1  0.4484938 0.07845312    398
   Mass Hsc82_phys Gravy.Score Half.life Instability.Index
1 47123          0       -0.79      10.8             41.68
2 47123          0       -0.79      10.8             41.68
3 47123          0       -0.79      10.8             41.68
4 47123          0       -0.79      10.8             41.68
5 47123          0       -0.79      10.8             41.68
6 47123          0       -0.79      10.8             41.68
  Protein.Abundance.Median.Absolute.Deviation Aliphatic.Index
1                                        1811           81.18
2                                        1811           81.18
3                                        1811           81.18
4                                        1811           81.18
5                                        1811           81.18
6                                        1811           81.18</code></pre>
</div>
</div>
</section>
<section id="data-collection" class="level1">
<h1>Data Collection</h1>
<p>The â€˜proteomicâ€™ data are the processed results of Data Independent Acquisition Mass Spectrometry experiments that quantified the relative protein abundances in cell lysates of mutant yeast compared to those of wild-type yeast. After processing the signal intensities of the 2 spectra (mutant vs wildtype), the calculated measurement is presented as Log base 2 Fold Change. Resources and information regarding proteins detected in our experiments are available at <a href="https://www.yeastgenome.org">Saccharomyces Genome Database</a>.</p>
</section>
<section id="summary-of-proteomic-data" class="level1">
<h1>Summary of Proteomic Data</h1>
<p>An advantage of the glimpse command is a neater output of the dataframe dimensions, column names, column class type, and â€˜sampleâ€™ view of the actual data in the columns.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">glimpse</span>(Proteomic.Data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Rows: 22,338
Columns: 14
$ Systemic.name                               &lt;chr&gt; "Q0140", "Q0140", "Q0140",~
$ contrast                                    &lt;chr&gt; "WTHsc82_vs_A583T", "WTHsc~
$ UniprotID                                   &lt;chr&gt; "P02381", "P02381", "P0238~
$ Gene                                        &lt;chr&gt; "VAR1", "VAR1", "VAR1", "V~
$ logFC                                       &lt;dbl&gt; -0.32719293, 0.31348064, 0~
$ P.Value                                     &lt;dbl&gt; 1.917392e-01, 2.102941e-01~
$ Length                                      &lt;int&gt; 398, 398, 398, 398, 398, 3~
$ Mass                                        &lt;int&gt; 47123, 47123, 47123, 47123~
$ Hsc82_phys                                  &lt;int&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0,~
$ Gravy.Score                                 &lt;dbl&gt; -0.79, -0.79, -0.79, -0.79~
$ Half.life                                   &lt;dbl&gt; 10.8, 10.8, 10.8, 10.8, 10~
$ Instability.Index                           &lt;dbl&gt; 41.68, 41.68, 41.68, 41.68~
$ Protein.Abundance.Median.Absolute.Deviation &lt;int&gt; 1811, 1811, 1811, 1811, 18~
$ Aliphatic.Index                             &lt;dbl&gt; 81.18, 81.18, 81.18, 81.18~</code></pre>
</div>
</div>
</section>
<section id="data-dictionary" class="level1">
<h1>Data Dictionary</h1>
<p>What are these columns and what do the values contained in them even represent? What on earth is a â€˜Gravy.Scoreâ€™? Fortunately, I have created a legend containing variable (attribute) descriptions.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>Attributes <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"MyData/Column_Descriptions.csv"</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(Attributes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<colgroup>
<col style="width: 17%">
<col style="width: 4%">
<col style="width: 6%">
<col style="width: 71%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Column.names</th>
<th style="text-align: left;">Data.type</th>
<th style="text-align: left;">Attribute.type</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Systemic.name</td>
<td style="text-align: left;">Item</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Unique ID for gene</td>
</tr>
<tr class="even">
<td style="text-align: left;">contrast</td>
<td style="text-align: left;">Attribute</td>
<td style="text-align: left;">Categorical</td>
<td style="text-align: left;">Pair-wise comparison between wild type and each yeast mutant</td>
</tr>
<tr class="odd">
<td style="text-align: left;">UniprotID</td>
<td style="text-align: left;">Item</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Unique ID for gene</td>
</tr>
<tr class="even">
<td style="text-align: left;">Gene</td>
<td style="text-align: left;">Item</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">Common name given to gene</td>
</tr>
<tr class="odd">
<td style="text-align: left;">logFC</td>
<td style="text-align: left;">Attribute</td>
<td style="text-align: left;">Quantitative</td>
<td style="text-align: left;">Relative protein abundance between mutant yeast to wildtype on log base 2 scale; negative values signify increase in abundance and positive values signify decrease in abundace</td>
</tr>
<tr class="even">
<td style="text-align: left;">P.Value</td>
<td style="text-align: left;">Attribute</td>
<td style="text-align: left;">Quantitative</td>
<td style="text-align: left;">Stastical significance of measurements (3 biological replicates)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Length</td>
<td style="text-align: left;">Attribute</td>
<td style="text-align: left;">Quantitative</td>
<td style="text-align: left;">The number of amino acid residues in protein sequence</td>
</tr>
<tr class="even">
<td style="text-align: left;">Mass</td>
<td style="text-align: left;">Attribute</td>
<td style="text-align: left;">Quantitative</td>
<td style="text-align: left;">the calculated mass of protein sequence</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Hsc82_phys</td>
<td style="text-align: left;">Attribute</td>
<td style="text-align: left;">Quantitative</td>
<td style="text-align: left;">If it is known to physically interact with Hsc82 protein; 1 means physical interaction; 0 means no physical interaction</td>
</tr>
<tr class="even">
<td style="text-align: left;">Gravy Score</td>
<td style="text-align: left;">Attribute</td>
<td style="text-align: left;">Quantitative</td>
<td style="text-align: left;">Is a estimation of how hydrophobic a protein is overall, more positive equals more hydrophobic, but does not take into account 3D structure.</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Half-life</td>
<td style="text-align: left;">Attribute</td>
<td style="text-align: left;">Quantitative</td>
<td style="text-align: left;">The time it takes in minutes for the cell to turn-over/degrade half the protein population</td>
</tr>
<tr class="even">
<td style="text-align: left;">Instability Index</td>
<td style="text-align: left;">Attribute</td>
<td style="text-align: left;">Quantitative</td>
<td style="text-align: left;">Estimation whether protein is stable in a test tube, values less than 40 are more stable than proteins with values greater than 40</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Protein-Abundance-Median-Absolute-Deviation</td>
<td style="text-align: left;">Attribute</td>
<td style="text-align: left;">Quantitative</td>
<td style="text-align: left;">Calculated estimate of protein molecules per cell</td>
</tr>
<tr class="even">
<td style="text-align: left;">Aliphatic Index</td>
<td style="text-align: left;">Attribute</td>
<td style="text-align: left;">Quantitative</td>
<td style="text-align: left;">The amount of volume occupied by aliphatic side chains such as Alanine, valine, isoleucine, and leucine; greater values may be indicative of thermostability.</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="st-action-target-pair-searchexplore---extremes" class="level1">
<h1>1st Action-Target pair: Search:Explore - Extremes</h1>
<p>Now that we have an understanding of the data we are working with. It is time to start using this data to answer a variety of questions related to my research and field. One of the most well known things in my field, is that perturbing Hsp90 function could affect the steady state levels of gene products (proteins) directly and indirectly. Therefore, first question is:</p>
<section id="how-many-if-any-proteins-had-a-significant-change-in-their-steady-state-levels-in-at-least-one-mutant-condition" class="level2">
<h2 class="anchored" data-anchor-id="how-many-if-any-proteins-had-a-significant-change-in-their-steady-state-levels-in-at-least-one-mutant-condition">How many, if any, proteins had a significant change in their steady-state levels in at least one mutant condition?</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>Gene <span class="ot">&lt;-</span> <span class="fu">unique</span>(Proteomic.Data<span class="sc">$</span>Gene[<span class="fu">abs</span>(Proteomic.Data<span class="sc">$</span>logFC) <span class="sc">&gt;=</span> <span class="fl">1.5</span> <span class="sc">&amp;</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>                                     Proteomic.Data<span class="sc">$</span>P.Value <span class="sc">&lt;=</span> <span class="fl">0.055</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>Hits <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">length</span>(Gene), <span class="st">"out of"</span>, <span class="fu">length</span>(<span class="fu">unique</span>(Proteomic.Data<span class="sc">$</span>Gene)), <span class="st">"proteins were significantly affected in at least one mutant condition; Here we consider any protein whose steady state level changed by at least 1.5 (|Log2FC|)"</span> )</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(Hits)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<colgroup>
<col style="width: 100%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">350 out of 2482 proteins were significantly affected in at least one mutant condition; Here we consider any protein whose steady state level changed by at least 1.5 (|Log2FC|)</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Considering we found these genes of interest using the attributes â€˜logFCâ€™ and â€˜P.Valueâ€™; lets use these attributes to visualize our findings.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> Proteomic.Data<span class="sc">$</span>logFC</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(Proteomic.Data<span class="sc">$</span>P.Value)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>group <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">1</span>, <span class="fu">nrow</span>(Proteomic.Data))               <span class="co"># Create group variable</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>group[<span class="fu">abs</span>(x) <span class="sc">&gt;=</span> <span class="fl">1.5</span> <span class="sc">&amp;</span> y <span class="sc">&gt;=</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.055</span>)] <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>group[<span class="fu">abs</span>(x) <span class="sc">&gt;=</span> <span class="fl">1.5</span> <span class="sc">&amp;</span> y <span class="sc">&lt;=</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.055</span>)] <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(x, y, </span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>     <span class="at">col =</span> group, </span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>     <span class="at">pch =</span> <span class="dv">20</span>,</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">25</span>),</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">6.5</span>, <span class="fl">6.5</span>),</span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="fu">expression</span>(<span class="st">"Log"</span>[<span class="dv">2</span>]<span class="sc">*</span><span class="st">"Fold Change"</span>),</span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="fu">expression</span>(<span class="st">"-Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">"(P.Value)"</span>)</span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v =</span> <span class="fu">c</span>(<span class="fl">1.5</span>, <span class="sc">-</span><span class="fl">1.5</span>), <span class="at">col =</span> <span class="st">"red"</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.055</span>), <span class="at">col =</span> <span class="st">"red"</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(<span class="at">x =</span> <span class="fl">1.45</span>, <span class="at">y =</span> <span class="dv">24</span>, <span class="at">labels =</span> <span class="st">"Decreased steady state levels </span><span class="sc">\n</span><span class="st"> p value &lt; 0. 55"</span>,</span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>     <span class="at">pos =</span> <span class="dv">4</span>, <span class="at">offset =</span> .<span class="dv">5</span>)</span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(<span class="at">x =</span> <span class="sc">-</span><span class="fl">1.45</span>, <span class="at">y =</span> <span class="dv">24</span>, <span class="at">labels =</span> <span class="st">"Increased steady state levels </span><span class="sc">\n</span><span class="st"> p value &lt; 0. 55"</span>,</span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a>     <span class="at">pos =</span> <span class="dv">2</span>, <span class="at">offset =</span> .<span class="dv">5</span>)</span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(<span class="at">x =</span> <span class="sc">-</span><span class="dv">4</span>, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.055</span>), <span class="at">labels =</span> <span class="st">"p value &gt; 0.055"</span>, <span class="at">pos =</span> <span class="dv">1</span>, <span class="at">offset =</span> .<span class="dv">5</span>)</span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(<span class="at">x =</span> <span class="dv">4</span>, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.055</span>), <span class="at">labels =</span> <span class="st">"p value &gt; 0.055"</span>, <span class="at">pos =</span> <span class="dv">1</span>, <span class="at">offset =</span> .<span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="index_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>With this â€˜volcanoâ€™ plot we visualized what was done prior when we searched for any proteins that met our criteria for significance. Those that met our criteria were marked pink and those that did not meet our criteria were marked black or light green. To better visualize the â€˜barriersâ€™ for significance, red-dashed lines were added. Proteins with positive logFC values had decreased levels and those with negative logFC values had increased levels. Although not relevant for future analysis, we can spot proteins with large changes in steady state levels, but unfortunately did not meet our threshold for statistical significance (light green) and this is was due to too much variance between the 3 biological replicates in the measurement of that protein.</p>


<!-- -->

</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb10" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Post With Code"</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Erick Rios"</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "2023-02-06"</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="an">categories:</span><span class="co"> [news, code, analysis]</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="an">image:</span><span class="co"> "image.jpg"</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="an">code-overflow:</span><span class="co"> wrap</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="an">code-tools:</span><span class="co"> true</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>::: callout-note</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>Nice work in general. Some minor editing of your section titles and some of the text. Added a hyperlink to the relevant online database. You've done a good job adding explanation to the code. A new person could read this and understand (and use) the data.</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a><span class="fu"># Goal of Assignment 2</span></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>In the quest to ensure data reproducibility, my goals are to:</span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a><span class="ss">1.  </span>Collect data of interest.</span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a><span class="ss">2.  </span>Import my data of interest into Rstudio.</span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a><span class="ss">3.  </span>Describe the data.</span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a><span class="ss">4.  </span>If time permits, host the data on a repository; probably github.</span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-27"><a href="#cb10-27" aria-hidden="true" tabindex="-1"></a><span class="co">#| output: false</span></span>
<span id="cb10-28"><a href="#cb10-28" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb10-29"><a href="#cb10-29" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-30"><a href="#cb10-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-31"><a href="#cb10-31" aria-hidden="true" tabindex="-1"></a><span class="fu"># Importation of 'Proteomic' Data</span></span>
<span id="cb10-32"><a href="#cb10-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-35"><a href="#cb10-35" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-36"><a href="#cb10-36" aria-hidden="true" tabindex="-1"></a>Proteomic.Data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"MyData/2022-03-21_trim-stacked.csv"</span>)</span>
<span id="cb10-37"><a href="#cb10-37" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(Proteomic.Data)</span>
<span id="cb10-38"><a href="#cb10-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-39"><a href="#cb10-39" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-40"><a href="#cb10-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-41"><a href="#cb10-41" aria-hidden="true" tabindex="-1"></a><span class="fu"># Data Collection</span></span>
<span id="cb10-42"><a href="#cb10-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-43"><a href="#cb10-43" aria-hidden="true" tabindex="-1"></a>The 'proteomic' data are the processed results of Data Independent Acquisition Mass Spectrometry experiments that quantified the relative protein abundances in cell lysates of mutant yeast compared to those of wild-type yeast. After processing the signal intensities of the 2 spectra (mutant vs wildtype), the calculated measurement is presented as Log base 2 Fold Change. Resources and information regarding proteins detected in our experiments are available at <span class="co">[</span><span class="ot">Saccharomyces Genome Database</span><span class="co">](https://www.yeastgenome.org)</span>.</span>
<span id="cb10-44"><a href="#cb10-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-45"><a href="#cb10-45" aria-hidden="true" tabindex="-1"></a><span class="fu"># Summary of Proteomic Data</span></span>
<span id="cb10-46"><a href="#cb10-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-47"><a href="#cb10-47" aria-hidden="true" tabindex="-1"></a>An advantage of the glimpse command is a neater output of the dataframe dimensions, column names, column class type, and 'sample' view of the actual data in the columns.</span>
<span id="cb10-48"><a href="#cb10-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-51"><a href="#cb10-51" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-52"><a href="#cb10-52" aria-hidden="true" tabindex="-1"></a><span class="fu">glimpse</span>(Proteomic.Data)</span>
<span id="cb10-53"><a href="#cb10-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-54"><a href="#cb10-54" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-55"><a href="#cb10-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-56"><a href="#cb10-56" aria-hidden="true" tabindex="-1"></a><span class="fu"># Data Dictionary</span></span>
<span id="cb10-57"><a href="#cb10-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-58"><a href="#cb10-58" aria-hidden="true" tabindex="-1"></a>What are these columns and what do the values contained in them even represent? What on earth is a 'Gravy.Score'? Fortunately, I have created a legend containing variable (attribute) descriptions.</span>
<span id="cb10-59"><a href="#cb10-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-62"><a href="#cb10-62" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-63"><a href="#cb10-63" aria-hidden="true" tabindex="-1"></a>Attributes <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"MyData/Column_Descriptions.csv"</span>)</span>
<span id="cb10-64"><a href="#cb10-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-65"><a href="#cb10-65" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(Attributes)</span>
<span id="cb10-66"><a href="#cb10-66" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-67"><a href="#cb10-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-68"><a href="#cb10-68" aria-hidden="true" tabindex="-1"></a><span class="fu"># 1st Action-Target pair: Search:Explore - Extremes</span></span>
<span id="cb10-69"><a href="#cb10-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-70"><a href="#cb10-70" aria-hidden="true" tabindex="-1"></a>Now that we have an understanding of the data we are working with. It is time to start using this data to answer a variety of questions related to my research and field. One of the most well known things in my field, is that perturbing Hsp90 function could affect the steady state levels of gene products (proteins) directly and indirectly. Therefore, first question is:</span>
<span id="cb10-71"><a href="#cb10-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-72"><a href="#cb10-72" aria-hidden="true" tabindex="-1"></a><span class="fu">## How many, if any, proteins had a significant change in their steady-state levels in at least one mutant condition?</span></span>
<span id="cb10-73"><a href="#cb10-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-76"><a href="#cb10-76" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-77"><a href="#cb10-77" aria-hidden="true" tabindex="-1"></a>Gene <span class="ot">&lt;-</span> <span class="fu">unique</span>(Proteomic.Data<span class="sc">$</span>Gene[<span class="fu">abs</span>(Proteomic.Data<span class="sc">$</span>logFC) <span class="sc">&gt;=</span> <span class="fl">1.5</span> <span class="sc">&amp;</span></span>
<span id="cb10-78"><a href="#cb10-78" aria-hidden="true" tabindex="-1"></a>                                     Proteomic.Data<span class="sc">$</span>P.Value <span class="sc">&lt;=</span> <span class="fl">0.055</span>])</span>
<span id="cb10-79"><a href="#cb10-79" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-80"><a href="#cb10-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-83"><a href="#cb10-83" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-84"><a href="#cb10-84" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb10-85"><a href="#cb10-85" aria-hidden="true" tabindex="-1"></a>Hits <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">length</span>(Gene), <span class="st">"out of"</span>, <span class="fu">length</span>(<span class="fu">unique</span>(Proteomic.Data<span class="sc">$</span>Gene)), <span class="st">"proteins were significantly affected in at least one mutant condition; Here we consider any protein whose steady state level changed by at least 1.5 (|Log2FC|)"</span> )</span>
<span id="cb10-86"><a href="#cb10-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-87"><a href="#cb10-87" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(Hits)</span>
<span id="cb10-88"><a href="#cb10-88" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-89"><a href="#cb10-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-90"><a href="#cb10-90" aria-hidden="true" tabindex="-1"></a>Considering we found these genes of interest using the attributes 'logFC' and 'P.Value'; lets use these attributes to visualize our findings.</span>
<span id="cb10-91"><a href="#cb10-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-92"><a href="#cb10-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-95"><a href="#cb10-95" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-96"><a href="#cb10-96" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb10-97"><a href="#cb10-97" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> Proteomic.Data<span class="sc">$</span>logFC</span>
<span id="cb10-98"><a href="#cb10-98" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(Proteomic.Data<span class="sc">$</span>P.Value)</span>
<span id="cb10-99"><a href="#cb10-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-100"><a href="#cb10-100" aria-hidden="true" tabindex="-1"></a>group <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">1</span>, <span class="fu">nrow</span>(Proteomic.Data))               <span class="co"># Create group variable</span></span>
<span id="cb10-101"><a href="#cb10-101" aria-hidden="true" tabindex="-1"></a>group[<span class="fu">abs</span>(x) <span class="sc">&gt;=</span> <span class="fl">1.5</span> <span class="sc">&amp;</span> y <span class="sc">&gt;=</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.055</span>)] <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb10-102"><a href="#cb10-102" aria-hidden="true" tabindex="-1"></a>group[<span class="fu">abs</span>(x) <span class="sc">&gt;=</span> <span class="fl">1.5</span> <span class="sc">&amp;</span> y <span class="sc">&lt;=</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.055</span>)] <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb10-103"><a href="#cb10-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-104"><a href="#cb10-104" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(x, y, </span>
<span id="cb10-105"><a href="#cb10-105" aria-hidden="true" tabindex="-1"></a>     <span class="at">col =</span> group, </span>
<span id="cb10-106"><a href="#cb10-106" aria-hidden="true" tabindex="-1"></a>     <span class="at">pch =</span> <span class="dv">20</span>,</span>
<span id="cb10-107"><a href="#cb10-107" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">25</span>),</span>
<span id="cb10-108"><a href="#cb10-108" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">6.5</span>, <span class="fl">6.5</span>),</span>
<span id="cb10-109"><a href="#cb10-109" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="fu">expression</span>(<span class="st">"Log"</span>[<span class="dv">2</span>]<span class="sc">*</span><span class="st">"Fold Change"</span>),</span>
<span id="cb10-110"><a href="#cb10-110" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="fu">expression</span>(<span class="st">"-Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">"(P.Value)"</span>)</span>
<span id="cb10-111"><a href="#cb10-111" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-112"><a href="#cb10-112" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v =</span> <span class="fu">c</span>(<span class="fl">1.5</span>, <span class="sc">-</span><span class="fl">1.5</span>), <span class="at">col =</span> <span class="st">"red"</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span>
<span id="cb10-113"><a href="#cb10-113" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.055</span>), <span class="at">col =</span> <span class="st">"red"</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span>
<span id="cb10-114"><a href="#cb10-114" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(<span class="at">x =</span> <span class="fl">1.45</span>, <span class="at">y =</span> <span class="dv">24</span>, <span class="at">labels =</span> <span class="st">"Decreased steady state levels </span><span class="sc">\n</span><span class="st"> p value &lt; 0. 55"</span>,</span>
<span id="cb10-115"><a href="#cb10-115" aria-hidden="true" tabindex="-1"></a>     <span class="at">pos =</span> <span class="dv">4</span>, <span class="at">offset =</span> .<span class="dv">5</span>)</span>
<span id="cb10-116"><a href="#cb10-116" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(<span class="at">x =</span> <span class="sc">-</span><span class="fl">1.45</span>, <span class="at">y =</span> <span class="dv">24</span>, <span class="at">labels =</span> <span class="st">"Increased steady state levels </span><span class="sc">\n</span><span class="st"> p value &lt; 0. 55"</span>,</span>
<span id="cb10-117"><a href="#cb10-117" aria-hidden="true" tabindex="-1"></a>     <span class="at">pos =</span> <span class="dv">2</span>, <span class="at">offset =</span> .<span class="dv">5</span>)</span>
<span id="cb10-118"><a href="#cb10-118" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(<span class="at">x =</span> <span class="sc">-</span><span class="dv">4</span>, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.055</span>), <span class="at">labels =</span> <span class="st">"p value &gt; 0.055"</span>, <span class="at">pos =</span> <span class="dv">1</span>, <span class="at">offset =</span> .<span class="dv">5</span>)</span>
<span id="cb10-119"><a href="#cb10-119" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(<span class="at">x =</span> <span class="dv">4</span>, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.055</span>), <span class="at">labels =</span> <span class="st">"p value &gt; 0.055"</span>, <span class="at">pos =</span> <span class="dv">1</span>, <span class="at">offset =</span> .<span class="dv">5</span>)</span>
<span id="cb10-120"><a href="#cb10-120" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-121"><a href="#cb10-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-122"><a href="#cb10-122" aria-hidden="true" tabindex="-1"></a>With this 'volcano' plot we visualized what was done prior when we searched for any proteins that met our criteria for significance. Those that met our criteria were marked pink and those that did not meet our criteria were marked black or light green. To better visualize the 'barriers' for significance, red-dashed lines were added. Proteins with positive logFC values had decreased levels and those with negative logFC values had increased levels. Although not relevant for future analysis, we can spot proteins with large changes in steady state levels, but unfortunately did not meet our threshold for statistical significance (light green) and this is was due to too much variance between the 3 biological replicates in the measurement of that protein.</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>