<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Erick Rios">
<meta name="dcterms.date" content="2023-05-03">
<meta name="description" content="Network visualization and analysis of proteomic responses to Hsp90 mutation.">

<title>Yeast, yeast, and more yeast. - BCB 520 - The Final Project</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../../styles.css">
</head>

<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Yeast, yeast, and more yeast.</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../about.html">
 <span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/"><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com"><i class="bi bi-twitter" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <div class="quarto-title-block"><div><h1 class="title">BCB 520 - The Final Project</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
            <p class="subtitle lead">Mutations in Molecular Chaperone: Hsp90 Elicit Different Proteomic Responses</p>
                  <div>
        <div class="description">
          Network visualization and analysis of proteomic responses to Hsp90 mutation.
        </div>
      </div>
                          <div class="quarto-categories">
                <div class="quarto-category">Portfolio</div>
                <div class="quarto-category">DataViz</div>
                <div class="quarto-category">Network</div>
                <div class="quarto-category">Project</div>
                <div class="quarto-category">Hsp90</div>
                <div class="quarto-category">Mutations</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Erick Rios </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">May 3, 2023</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(igraph)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(viridis)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="the-preamble" class="level1">
<h1>The Preamble</h1>
<p>The focus of my research is the study of Hsp90 molecular chaperone. Over the decades we come to realize this chaperone sits as a major hub of protein-protein interactions <span class="citation" data-cites="Gopinath2014 Zhao2005a Rizzolo2019">(<a href="#ref-Gopinath2014" role="doc-biblioref">Gopinath et al. 2014</a>; <a href="#ref-Zhao2005a" role="doc-biblioref">Zhao et al. 2005</a>; <a href="#ref-Rizzolo2019" role="doc-biblioref">Rizzolo and Houry 2019</a>)</span> in which it regulates the folding, activity, and degradation of its clients.</p>
<p>The current study expands the preliminary proteomic analysis previously done in yeast by us<span class="citation" data-cites="Hohrman2021">(<a href="#ref-Hohrman2021" role="doc-biblioref">Hohrman et al. 2021</a>)</span>, here we use quantitative proteomic analysis to compare extracts from yeast expressing wild-type yeast Hsp90 to nine different mutants. Our results demonstrate that the mutants elicit varied proteomic responses, even though the cells were harvested at temperatures that do not significantly affect growth. Overall, our results suggests:</p>
<ul>
<li><strong>cellular processes</strong> that may be differentially affected by Hsc82 mutation.</li>
</ul>
</section>
<section id="data" class="level1">
<h1>Data</h1>
<section id="significant-hits-in-our-proteomic-data-from-dia-ms" class="level2">
<h2 class="anchored" data-anchor-id="significant-hits-in-our-proteomic-data-from-dia-ms">Significant hits in our Proteomic Data from DIA-MS</h2>
<p>Data source:</p>
<ol type="1">
<li>Main data used in this analysis are the results of Data Independent Acquisition Mass Spectrometry (DIA-MS) experiments, a technique for quantitative proteomics. This is the crux of our study. For brevity, the data has been pre-filtered for significant hits of interest for analysis.</li>
</ol>
<p>A significant hit in our study is a protein whose abundance had a |Log<sub>2</sub> Fold Change| (LogFC) greater than 1.5, relative to wild-type.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load in our data</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>proteomic <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"trimmed_proteomic_Data.csv"</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">glimpse</span>(proteomic)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Rows: 536
Columns: 7
$ Index         &lt;int&gt; 978, 1096, 1948, 2981, 3981, 4064, 4084, 4153, 4177, 590~
$ Systemic.name &lt;chr&gt; "YBR082C", "YBR102C", "YCL047C", "YDL181W", "YDR214W", "~
$ contrast      &lt;chr&gt; "A583T", "A583T", "A583T", "A583T", "A583T", "A583T", "A~
$ UniprotID     &lt;chr&gt; "P15731", "P38261", "P25576", "P01097", "Q12449", "Q0494~
$ Gene          &lt;chr&gt; "UBC4", "EXO84", "POF1", "INH1", "AHA1", "RTN1", "FMN1",~
$ logFC         &lt;dbl&gt; 2.800538, 1.707334, -5.045534, -1.636702, -1.893020, -1.~
$ P.Value       &lt;dbl&gt; 1.777647e-03, 2.010339e-02, 3.890000e-05, 2.590000e-06, ~</code></pre>
</div>
</div>
<section id="proteomic-data-dictionary" class="level3">
<h3 class="anchored" data-anchor-id="proteomic-data-dictionary">Proteomic Data Dictionary</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>proteomic.dictionary <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"Proteomic_data_dictionary.csv"</span>)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(proteomic.dictionary)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<colgroup>
<col style="width: 7%">
<col style="width: 92%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Attribute</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Index</td>
<td style="text-align: left;">Old indexing from the original entire data set, artifact from “write.csv” function</td>
</tr>
<tr class="even">
<td style="text-align: left;">Systemic.name</td>
<td style="text-align: left;">Unique identifiers for genes in yeast</td>
</tr>
<tr class="odd">
<td style="text-align: left;">contrast</td>
<td style="text-align: left;">Our single point mutations in Hsc82 isoform; This is sometimes referred to as our ‘experimental conditions’</td>
</tr>
<tr class="even">
<td style="text-align: left;">UniprotID</td>
<td style="text-align: left;">Unique identifiers for genes</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Gene</td>
<td style="text-align: left;">Gene symbol; symbols to genes by researchers and voted on by committees</td>
</tr>
<tr class="even">
<td style="text-align: left;">logFC</td>
<td style="text-align: left;">Log Fold Change in protein abundance in mutant sample over wild type sample; Log base 2; negative values signify increase in abundance, positive values is decreased abundance.</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P.Value</td>
<td style="text-align: left;">corrected P value using Bonferroni correction</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
</section>
<section id="list-of-hsp90-interactions" class="level2">
<h2 class="anchored" data-anchor-id="list-of-hsp90-interactions">List of Hsp90 Interactions</h2>
<ol start="2" type="1">
<li>Supporting data/information publicly available on <em>Saccharomyces Genome Database (<a href="https://yeastgenome.org/" title="Saccharomyces Genome Database main page">SGD</a>)</em></li>
</ol>
<ul>
<li>List of currently known protein <strong>physical</strong> interactors of Hsp90 (Both <a href="https://yeastgenome.org/locus/S000004798/interaction" title="List of Hsc82 genetic and physical interactors">Hsc82</a> and <a href="https://yeastgenome.org/locus/S000006161/interaction" title="List of Hsp82 genetic and physical interactors">Hsp82</a> isoforms)</li>
<li>List of currently known protein <strong>genetic</strong> interactors of Hsp90 (Both <a href="https://yeastgenome.org/locus/S000004798/interaction" title="List of Hsc82 genetic and physical interactors">Hsc82</a> and <a href="https://yeastgenome.org/locus/S000006161/interaction" title="List of Hsp82 genetic and physical interactors">Hsp82</a> isoforms)</li>
</ul>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>Hsp90.Interactors <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"SGD_All_interactions_Hsp90_with_Refs_trimmed.csv"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">glimpse</span>(Hsp90.Interactors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Rows: 4,990
Columns: 7
$ TYPE                  &lt;chr&gt; "Genetic", "Genetic", "Genetic", "Genetic", "Gen~
$ HSP90                 &lt;chr&gt; "HSC82", "HSC82", "HSC82", "HSC82", "HSC82", "HS~
$ HSP90.Systematic.Name &lt;chr&gt; "YMR186W", "YMR186W", "YMR186W", "YMR186W", "YMR~
$ Gene                  &lt;chr&gt; "AAH1", "AAR2", "AAR2", "ABD1", "ABD1", "ABD1", ~
$ Systemic.Name         &lt;chr&gt; "YNL141W", "YBL074C", "YBL074C", "YBR236C", "YBR~
$ Source                &lt;chr&gt; "BioGRID", "BioGRID", "BioGRID", "BioGRID", "Bio~
$ Reference             &lt;chr&gt; "Choi J, et al. (2017) PMID:28520786", "Costanzo~</code></pre>
</div>
</div>
<section id="hsp90-interaction-dictionary" class="level3">
<h3 class="anchored" data-anchor-id="hsp90-interaction-dictionary">Hsp90 Interaction Dictionary</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>Hsp90.Interactors.dictionary <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"SGD_Hsp90_interactions_dictionary.csv"</span>)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(Hsp90.Interactors.dictionary)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<colgroup>
<col style="width: 13%">
<col style="width: 86%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Attribute</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">TYPE</td>
<td style="text-align: left;">Type of interaction with either Hsc82 or Hsp82 isoforms of Hsp90; either Genetic or Physical</td>
</tr>
<tr class="even">
<td style="text-align: left;">HSP90</td>
<td style="text-align: left;">Gene symbol for either isoform: Hsc82 or Hsp82</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HSP90 Systematic Name</td>
<td style="text-align: left;">Yeast systematic identifier for isoforms Hsc82 or Hsp82naming convention</td>
</tr>
<tr class="even">
<td style="text-align: left;">Gene</td>
<td style="text-align: left;">Gene symbol for all genes/proteins currently known to have either genetic or physical interaction with either for Hsp90 isoform.</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Systemic Name</td>
<td style="text-align: left;">Yeast systematic identifiers for all genes/proteins currently known to have either genetic or physical interaction with either for Hsp90 isoform.</td>
</tr>
<tr class="even">
<td style="text-align: left;">Source</td>
<td style="text-align: left;">Database accessed by SGD to aggregate list</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Reference</td>
<td style="text-align: left;">References to studies that provided interaction data</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
</section>
<section id="go-slim-mapper-results" class="level2">
<h2 class="anchored" data-anchor-id="go-slim-mapper-results">GO Slim Mapper results</h2>
<p>Hits from our data analyzed by <a href="https://yeastgenome.org/goSlimMapper" title="Gene Ontology tool to classify and group genes/proteins">Gene Ontology (GO) Slim Mapper</a></p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>GO.terms <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="at">file =</span> <span class="st">"GO Slim Mapper Results.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">glimpse</span>(GO.terms)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Rows: 89
Columns: 9
$ GOID                  &lt;chr&gt; "GO:0042221", "GO:0006520", "GO:0005975", "GO:00~
$ TERM                  &lt;chr&gt; "response to chemical", "cellular amino acid met~
$ NUM_LIST_ANNOTATIONS  &lt;int&gt; 35, 28, 27, 26, 21, 21, 20, 19, 16, 15, 14, 14, ~
$ LIST_SIZE             &lt;int&gt; 349, 349, 349, 349, 349, 349, 349, 349, 349, 349~
$ CLUSTER_FREQUENCY     &lt;chr&gt; "10.03%", "8.02%", "7.74%", "7.45%", "6.02%", "6~
$ TOTAL_NUM_ANNOTATIONS &lt;int&gt; 454, 155, 148, 163, 282, 106, 80, 127, 167, 271,~
$ POPULATION_SIZE       &lt;int&gt; 6485, 6485, 6485, 6485, 6485, 6485, 6485, 6485, ~
$ GENOME_FREQUENCY      &lt;chr&gt; "7.00%", "2.39%", "2.28%", "2.51%", "4.35%", "1.~
$ ANNOTATED_GENES       &lt;chr&gt; "ADD37, AHP1, CTT1, DAK2, DAL81, ECM38, FRA1, GA~</code></pre>
</div>
</div>
<section id="go-slim-mapper-results-dictionary" class="level3">
<h3 class="anchored" data-anchor-id="go-slim-mapper-results-dictionary">GO Slim Mapper results Dictionary</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>Go.dictionary <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"GO_process_results_dictionary.csv"</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(proteomic.dictionary)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<colgroup>
<col style="width: 7%">
<col style="width: 92%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Attribute</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Index</td>
<td style="text-align: left;">Old indexing from the original entire data set, artifact from “write.csv” function</td>
</tr>
<tr class="even">
<td style="text-align: left;">Systemic.name</td>
<td style="text-align: left;">Unique identifiers for genes in yeast</td>
</tr>
<tr class="odd">
<td style="text-align: left;">contrast</td>
<td style="text-align: left;">Our single point mutations in Hsc82 isoform; This is sometimes referred to as our ‘experimental conditions’</td>
</tr>
<tr class="even">
<td style="text-align: left;">UniprotID</td>
<td style="text-align: left;">Unique identifiers for genes</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Gene</td>
<td style="text-align: left;">Gene symbol; symbols to genes by researchers and voted on by committees</td>
</tr>
<tr class="even">
<td style="text-align: left;">logFC</td>
<td style="text-align: left;">Log Fold Change in protein abundance in mutant sample over wild type sample; Log base 2; negative values signify increase in abundance, positive values is decreased abundance.</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P.Value</td>
<td style="text-align: left;">corrected P value using Bonferroni correction</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
</section>
</section>
<section id="visualizations" class="level1">
<h1>Visualizations</h1>
<section id="figure-1" class="level2">
<h2 class="anchored" data-anchor-id="figure-1">Figure 1</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="do">### Trim and format the Gene Ontology: Process data</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate GO process enrichment</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>GO.terms<span class="sc">$</span>CLUSTER_FREQUENCY <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">sub</span>(<span class="st">"%"</span>, <span class="st">""</span>, GO.terms<span class="sc">$</span>CLUSTER_FREQUENCY))<span class="sc">/</span><span class="dv">100</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>GO.terms<span class="sc">$</span>GENOME_FREQUENCY <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">sub</span>(<span class="st">"%"</span>, <span class="st">""</span>, GO.terms<span class="sc">$</span>GENOME_FREQUENCY))<span class="sc">/</span><span class="dv">100</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>GO.terms<span class="sc">$</span>Enrichment <span class="ot">&lt;-</span> GO.terms<span class="sc">$</span>CLUSTER_FREQUENCY<span class="sc">/</span>GO.terms<span class="sc">$</span>GENOME_FREQUENCY</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Take top 5 enriched processes that have at least 5 proteins annotated to a cellular process.</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>GO.terms5 <span class="ot">&lt;-</span> GO.terms[GO.terms<span class="sc">$</span>NUM_LIST_ANNOTATIONS <span class="sc">&gt;=</span> <span class="dv">5</span>,]</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>GO.terms5 <span class="ot">&lt;-</span> GO.terms5[<span class="fu">order</span>(GO.terms5<span class="sc">$</span>Enrichment, <span class="at">decreasing =</span> T),]</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>GO.terms5 <span class="ot">&lt;-</span> GO.terms5[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">9</span>,<span class="dv">10</span>)]</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Format for network</span></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>GO.terms5.split <span class="ot">&lt;-</span> <span class="fu">separate_rows</span>(GO.terms5, ANNOTATED_GENES, <span class="at">sep =</span> <span class="st">", "</span>)</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a><span class="do">###</span></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Make initial network</span></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>vertices <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(GO.terms5.split<span class="sc">$</span>TERM, GO.terms5.split<span class="sc">$</span>ANNOTATED_GENES))</span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">graph_from_data_frame</span>(GO.terms5.split[,<span class="dv">2</span><span class="sc">:</span><span class="dv">3</span>], <span class="at">directed =</span> F, <span class="at">vertices =</span> vertices)</span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a><span class="do">###</span></span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a><span class="co"># create layout using force directed algorithm</span></span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a>layout <span class="ot">&lt;-</span> <span class="fu">layout_with_fr</span>(g, <span class="at">weights =</span> GO.terms5.split<span class="sc">$</span>Enrichment, <span class="at">coords =</span> <span class="fu">matrix</span>(<span class="at">data =</span> <span class="dv">0</span>, <span class="at">nrow =</span> <span class="dv">91</span>, <span class="at">ncol =</span> <span class="dv">2</span>))</span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a><span class="do">###</span></span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Add mutant nodes </span></span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true" tabindex="-1"></a>mutants <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">c</span>(<span class="fu">cos</span>(<span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">2</span><span class="sc">*</span>pi, <span class="at">length.out =</span> <span class="dv">10</span>)), </span>
<span id="cb11-31"><a href="#cb11-31" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">sin</span>(<span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">2</span><span class="sc">*</span>pi, <span class="at">length.out =</span> <span class="dv">10</span>))), <span class="at">ncol =</span> <span class="dv">2</span>)<span class="sc">*</span><span class="dv">11</span> <span class="co"># Create coordinates for a ring.</span></span>
<span id="cb11-32"><a href="#cb11-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-33"><a href="#cb11-33" aria-hidden="true" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">add_vertices</span>(g, <span class="dv">9</span>)</span>
<span id="cb11-34"><a href="#cb11-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-35"><a href="#cb11-35" aria-hidden="true" tabindex="-1"></a>mutant.names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"R46G"</span>, <span class="st">"G309S"</span>, <span class="st">"S481Y"</span>, <span class="st">"A583T"</span>, <span class="st">"K102E"</span>, <span class="st">"S25P"</span>, <span class="st">"Q380K"</span>, <span class="st">"G424D"</span>, <span class="st">"W296A"</span>)</span>
<span id="cb11-36"><a href="#cb11-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-37"><a href="#cb11-37" aria-hidden="true" tabindex="-1"></a><span class="fu">V</span>(g)<span class="sc">$</span>name[<span class="dv">92</span><span class="sc">:</span><span class="dv">100</span>] <span class="ot">&lt;-</span> mutant.names</span>
<span id="cb11-38"><a href="#cb11-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-39"><a href="#cb11-39" aria-hidden="true" tabindex="-1"></a><span class="co"># and edges between mutants and proteins/genes</span></span>
<span id="cb11-40"><a href="#cb11-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-41"><a href="#cb11-41" aria-hidden="true" tabindex="-1"></a>edgelist <span class="ot">&lt;-</span> proteomic[,<span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">5</span>)][proteomic<span class="sc">$</span>Gene <span class="sc">%in%</span> GO.terms5.split<span class="sc">$</span>ANNOTATED_GENES,]</span>
<span id="cb11-42"><a href="#cb11-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-43"><a href="#cb11-43" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert edge list to indices</span></span>
<span id="cb11-44"><a href="#cb11-44" aria-hidden="true" tabindex="-1"></a>edgelist<span class="sc">$</span>from <span class="ot">&lt;-</span> <span class="fu">match</span>(edgelist<span class="sc">$</span>contrast, <span class="fu">V</span>(g)<span class="sc">$</span>name)</span>
<span id="cb11-45"><a href="#cb11-45" aria-hidden="true" tabindex="-1"></a>edgelist<span class="sc">$</span>to <span class="ot">&lt;-</span> <span class="fu">match</span>(edgelist<span class="sc">$</span>Gene, <span class="fu">V</span>(g)<span class="sc">$</span>name)</span>
<span id="cb11-46"><a href="#cb11-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-47"><a href="#cb11-47" aria-hidden="true" tabindex="-1"></a><span class="co"># create a list of alternating values from 'from' and 'to' columns</span></span>
<span id="cb11-48"><a href="#cb11-48" aria-hidden="true" tabindex="-1"></a>edges <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rbind</span>(edgelist<span class="sc">$</span>from, edgelist<span class="sc">$</span>to))</span>
<span id="cb11-49"><a href="#cb11-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-50"><a href="#cb11-50" aria-hidden="true" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">add_edges</span>(g, <span class="at">edges =</span> edges)</span>
<span id="cb11-51"><a href="#cb11-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-52"><a href="#cb11-52" aria-hidden="true" tabindex="-1"></a><span class="do">###</span></span>
<span id="cb11-53"><a href="#cb11-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-54"><a href="#cb11-54" aria-hidden="true" tabindex="-1"></a><span class="co"># Now add the nodes for proteins that were either not in the top 5 enriched processes or could not be annotated to a GO process at all.</span></span>
<span id="cb11-55"><a href="#cb11-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-56"><a href="#cb11-56" aria-hidden="true" tabindex="-1"></a><span class="do">###</span></span>
<span id="cb11-57"><a href="#cb11-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-58"><a href="#cb11-58" aria-hidden="true" tabindex="-1"></a><span class="co"># Assign node colors based on physical, genetic, both (mixed), interaction with Hsp90 or neither (unknown)</span></span>
<span id="cb11-59"><a href="#cb11-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-60"><a href="#cb11-60" aria-hidden="true" tabindex="-1"></a>proteomic<span class="sc">$</span>Interaction <span class="ot">&lt;-</span> <span class="st">"Unknown"</span></span>
<span id="cb11-61"><a href="#cb11-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-62"><a href="#cb11-62" aria-hidden="true" tabindex="-1"></a>proteomic<span class="sc">$</span>Interaction[proteomic<span class="sc">$</span>Systemic.name <span class="sc">%in%</span> Hsp90.Interactors<span class="sc">$</span>Systemic.Name[Hsp90.Interactors<span class="sc">$</span>TYPE<span class="sc">==</span><span class="st">"Genetic"</span>]] <span class="ot">&lt;-</span> <span class="st">"Genetic"</span></span>
<span id="cb11-63"><a href="#cb11-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-64"><a href="#cb11-64" aria-hidden="true" tabindex="-1"></a>proteomic<span class="sc">$</span>Interaction[proteomic<span class="sc">$</span>Systemic.name <span class="sc">%in%</span> Hsp90.Interactors<span class="sc">$</span>Systemic.Name[Hsp90.Interactors<span class="sc">$</span>TYPE<span class="sc">==</span><span class="st">"Physical"</span>]] <span class="ot">&lt;-</span> <span class="st">"Physical"</span></span>
<span id="cb11-65"><a href="#cb11-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-66"><a href="#cb11-66" aria-hidden="true" tabindex="-1"></a>proteomic<span class="sc">$</span>Interaction[proteomic<span class="sc">$</span>Systemic.name <span class="sc">%in%</span> Hsp90.Interactors<span class="sc">$</span>Systemic.Name[Hsp90.Interactors<span class="sc">$</span>TYPE<span class="sc">==</span><span class="st">"Physical"</span>] <span class="sc">&amp;</span></span>
<span id="cb11-67"><a href="#cb11-67" aria-hidden="true" tabindex="-1"></a>                        proteomic<span class="sc">$</span>Systemic.name <span class="sc">%in%</span> Hsp90.Interactors<span class="sc">$</span>Systemic.Name[Hsp90.Interactors<span class="sc">$</span>TYPE<span class="sc">==</span><span class="st">"Genetic"</span>]] <span class="ot">&lt;-</span> <span class="st">"Mixed"</span></span>
<span id="cb11-68"><a href="#cb11-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-69"><a href="#cb11-69" aria-hidden="true" tabindex="-1"></a><span class="co"># Define color palette for the color blind plebs, just kidding, but not really</span></span>
<span id="cb11-70"><a href="#cb11-70" aria-hidden="true" tabindex="-1"></a>colors <span class="ot">&lt;-</span> <span class="fu">viridis</span>(<span class="at">n =</span> <span class="dv">4</span>, <span class="at">option =</span> <span class="st">"D"</span>, <span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">begin =</span> <span class="dv">0</span>, <span class="at">end =</span> <span class="dv">1</span>, <span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>)</span>
<span id="cb11-71"><a href="#cb11-71" aria-hidden="true" tabindex="-1"></a>colors.process <span class="ot">&lt;-</span> <span class="fu">viridis</span>(<span class="at">n =</span> <span class="dv">5</span>, <span class="at">option =</span> <span class="st">"F"</span>, <span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">begin =</span> <span class="dv">0</span>, <span class="at">end =</span> <span class="dv">1</span>, <span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>)</span>
<span id="cb11-72"><a href="#cb11-72" aria-hidden="true" tabindex="-1"></a>proteomic<span class="sc">$</span>Int.color <span class="ot">&lt;-</span> colors[<span class="fu">match</span>(proteomic<span class="sc">$</span>Interaction, <span class="fu">unique</span>(proteomic<span class="sc">$</span>Interaction))]</span>
<span id="cb11-73"><a href="#cb11-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-74"><a href="#cb11-74" aria-hidden="true" tabindex="-1"></a><span class="fu">V</span>(g)<span class="sc">$</span>color <span class="ot">&lt;-</span> <span class="fu">c</span>(colors.process, </span>
<span id="cb11-75"><a href="#cb11-75" aria-hidden="true" tabindex="-1"></a>                proteomic<span class="sc">$</span>Int.color[<span class="fu">match</span>(<span class="fu">V</span>(g)<span class="sc">$</span>name, proteomic<span class="sc">$</span>Gene)][<span class="dv">6</span><span class="sc">:</span><span class="dv">91</span>],</span>
<span id="cb11-76"><a href="#cb11-76" aria-hidden="true" tabindex="-1"></a>                <span class="fu">rep</span>(<span class="st">"white"</span>, <span class="dv">9</span>)</span>
<span id="cb11-77"><a href="#cb11-77" aria-hidden="true" tabindex="-1"></a>                )</span>
<span id="cb11-78"><a href="#cb11-78" aria-hidden="true" tabindex="-1"></a><span class="fu">V</span>(g)<span class="sc">$</span>color[<span class="dv">23</span>] <span class="ot">&lt;-</span> <span class="st">"#35B779FF"</span></span>
<span id="cb11-79"><a href="#cb11-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-80"><a href="#cb11-80" aria-hidden="true" tabindex="-1"></a><span class="do">###</span></span>
<span id="cb11-81"><a href="#cb11-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-82"><a href="#cb11-82" aria-hidden="true" tabindex="-1"></a><span class="co"># Modify node size, shape, and labels to display</span></span>
<span id="cb11-83"><a href="#cb11-83" aria-hidden="true" tabindex="-1"></a>term.count <span class="ot">&lt;-</span>  <span class="fu">length</span>(<span class="fu">unique</span>(GO.terms5.split<span class="sc">$</span>TERM))</span>
<span id="cb11-84"><a href="#cb11-84" aria-hidden="true" tabindex="-1"></a>node.count <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">unique</span>(GO.terms5.split<span class="sc">$</span>ANNOTATED_GENES))</span>
<span id="cb11-85"><a href="#cb11-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-86"><a href="#cb11-86" aria-hidden="true" tabindex="-1"></a><span class="fu">V</span>(g)<span class="sc">$</span>size <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="dv">12</span>, term.count), <span class="fu">rep</span>(<span class="dv">5</span>, node.count), <span class="fu">rep</span>(<span class="dv">0</span>, <span class="dv">9</span>))</span>
<span id="cb11-87"><a href="#cb11-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-88"><a href="#cb11-88" aria-hidden="true" tabindex="-1"></a><span class="fu">V</span>(g)<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">""</span>, <span class="dv">91</span>), mutant.names)</span>
<span id="cb11-89"><a href="#cb11-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-90"><a href="#cb11-90" aria-hidden="true" tabindex="-1"></a><span class="fu">V</span>(g)<span class="sc">$</span>shape <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"circle"</span>, term.count), <span class="fu">rep</span>(<span class="st">"square"</span> , node.count), <span class="fu">rep</span>(<span class="st">"none"</span>, <span class="dv">9</span>))</span>
<span id="cb11-91"><a href="#cb11-91" aria-hidden="true" tabindex="-1"></a><span class="do">###</span></span>
<span id="cb11-92"><a href="#cb11-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-93"><a href="#cb11-93" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(g,</span>
<span id="cb11-94"><a href="#cb11-94" aria-hidden="true" tabindex="-1"></a>     <span class="at">layout =</span> <span class="fu">rbind</span>(layout, mutants[<span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]),</span>
<span id="cb11-95"><a href="#cb11-95" aria-hidden="true" tabindex="-1"></a>     <span class="at">margin =</span> <span class="sc">-</span>.<span class="dv">8</span>, <span class="co"># allow the network to use up more of the margin</span></span>
<span id="cb11-96"><a href="#cb11-96" aria-hidden="true" tabindex="-1"></a>     <span class="at">vertex.label.cex =</span> <span class="dv">1</span>,</span>
<span id="cb11-97"><a href="#cb11-97" aria-hidden="true" tabindex="-1"></a>     <span class="at">vertex.label.font =</span> <span class="dv">2</span>, <span class="co"># Bold</span></span>
<span id="cb11-98"><a href="#cb11-98" aria-hidden="true" tabindex="-1"></a>     <span class="at">vertex.label.color=</span> <span class="st">"black"</span>,</span>
<span id="cb11-99"><a href="#cb11-99" aria-hidden="true" tabindex="-1"></a>     <span class="at">edge_arrow_mode =</span> <span class="dv">0</span>,</span>
<span id="cb11-100"><a href="#cb11-100" aria-hidden="true" tabindex="-1"></a>     <span class="at">vertex.color =</span> <span class="fu">V</span>(g)<span class="sc">$</span>color,</span>
<span id="cb11-101"><a href="#cb11-101" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="fl">1.75</span>), <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>, <span class="dv">2</span>) <span class="co"># Increase the plot size to help accommodate the network</span></span>
<span id="cb11-102"><a href="#cb11-102" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb11-103"><a href="#cb11-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-104"><a href="#cb11-104" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"topleft"</span>, <span class="at">legend=</span><span class="fu">unique</span>(proteomic<span class="sc">$</span>Interaction), <span class="at">fill=</span><span class="fu">unique</span>(<span class="fu">V</span>(g)<span class="sc">$</span>color)[<span class="dv">6</span><span class="sc">:</span><span class="dv">91</span>], <span class="at">title =</span> <span class="st">"Hsp90 Interaction Type"</span>, <span class="at">cex =</span> .<span class="dv">75</span>)</span>
<span id="cb11-105"><a href="#cb11-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-106"><a href="#cb11-106" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"bottomleft"</span>, <span class="at">legend=</span> <span class="fu">V</span>(g)<span class="sc">$</span>name[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>], <span class="at">title =</span> <span class="st">"GO Process"</span>, <span class="at">cex =</span> .<span class="dv">75</span>, <span class="at">pch =</span> <span class="dv">21</span>, <span class="at">pt.cex =</span> <span class="dv">2</span>, <span class="at">pt.bg =</span> <span class="fu">V</span>(g)<span class="sc">$</span>color[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>], <span class="at">pt.lwd =</span> <span class="fl">0.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="index_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<section id="nodes" class="level3">
<h3 class="anchored" data-anchor-id="nodes">NODES</h3>
<p>Here we have 3 classes or groups of nodes:</p>
<ol type="1">
<li>Our 9 different Hsc82 mutations</li>
<li>The top 5 most enriched cellular processes given by Gene Ontology Slim Mapper tool</li>
<li>86 different proteins/genes that were annotated to one or more of the top cellular processes <em>and</em> protein abundance was significantly affected in one or more Hsc82 mutants</li>
</ol>
<p>Node labels were removed for network clarity, with exception of mutant nodes. The GO process terms were moved to a legend and color was used to identify the GO terms in the figure. The focus of the figure was to identify potential processes, not individual proteins, that may be affected by one or more Hsc82 mutant, therefore labels for individual proteins were removed.</p>
<p>Node colors were picked in consideration for color blind individuals using the “Viridis” color package.</p>
<p>In an effort to help reduce cognitive demand, the node shapes and sizes were adjusted to help visually separate the node classes and direct the viewer from the outer layer of the network, towards the center with increasing detail. Node shape for the mutations were removed and labels were left as the main focus of our study revolves around these mutations. The GO terms were designated as circles and size is larger compared to the individual proteins which were designated as squares. Overall, the increase in shape size moving outward from the center gives a sense of hierarchy of importance.</p>
<p>By focusing on the top 5 enriched GO processes, the classic hairball problem is avoided and we can show with visual clarity the cellular processes that would most likely be considered for follow-up experiments or studies.</p>
</section>
<section id="edges" class="level3">
<h3 class="anchored" data-anchor-id="edges">EDGES</h3>
<p>Genes/proteins annotated to given GO terms by GO Slim mapper had edges drawn between the protein and the corresponding GO term(s). A few proteins, were annotated to more than one cellular process.</p>
<p>Edges were drawn between proteins and Hsc82 mutations if the protein’s abundance was significantly affected by an Hsc82 mutation. Some proteins were affected by more than one mutation and therefore have multiple edges.</p>
</section>
<section id="layout" class="level3">
<h3 class="anchored" data-anchor-id="layout">LAYOUT</h3>
<p>An initial network was made consisting of GO terms and proteins, after which the Fruchterman-Reingold force directed algorithm was employed and the GO term enrichments were used as edge weights.</p>
<p>Afterwards, the Hsc82 mutant nodes were manually added around the perimeter of the initial network to avoid mutant nodes from influencing the force directed graph and avoid a network hairball.</p>
</section>
</section>
</section>
<section id="conclusions" class="level1">
<h1>Conclusions</h1>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(g,</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>     <span class="at">layout =</span> <span class="fu">rbind</span>(layout, mutants[<span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]),</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>     <span class="at">margin =</span> <span class="sc">-</span>.<span class="dv">8</span>, <span class="co"># allow the network to use up more of the margin</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">vertex.label.cex =</span> <span class="dv">1</span>,</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">vertex.label.font =</span> <span class="dv">2</span>, <span class="co"># Bold</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">vertex.label.color=</span> <span class="st">"black"</span>,</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>     <span class="at">edge_arrow_mode =</span> <span class="dv">0</span>,</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>     <span class="at">vertex.color =</span> <span class="fu">V</span>(g)<span class="sc">$</span>color,</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="fl">1.75</span>), <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>, <span class="dv">2</span>) <span class="co"># Increase the plot size to help accommodate the network</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"topleft"</span>, <span class="at">legend=</span><span class="fu">unique</span>(proteomic<span class="sc">$</span>Interaction), <span class="at">fill=</span><span class="fu">unique</span>(<span class="fu">V</span>(g)<span class="sc">$</span>color)[<span class="dv">6</span><span class="sc">:</span><span class="dv">91</span>], <span class="at">title =</span> <span class="st">"Hsp90 Interaction Type"</span>, <span class="at">cex =</span> .<span class="dv">75</span>)</span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"bottomleft"</span>, <span class="at">legend=</span> <span class="fu">V</span>(g)<span class="sc">$</span>name[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>], <span class="at">title =</span> <span class="st">"GO Process"</span>, <span class="at">cex =</span> .<span class="dv">75</span>, <span class="at">pch =</span> <span class="dv">21</span>, <span class="at">pt.cex =</span> <span class="dv">2</span>, <span class="at">pt.bg =</span> <span class="fu">V</span>(g)<span class="sc">$</span>color[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>], <span class="at">pt.lwd =</span> <span class="fl">0.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="index_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Preliminary network analysis reveals</p>


<!-- -->


</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-Gopinath2014" class="csl-entry" role="doc-biblioentry">
Gopinath, Rajaneesh Karimpurath, Shu-Ting You, Kun-Yi Chien, Krishna B. S. Swamy, Jau-Song Yu, Scott C. Schuyler, and Jun-Yi Leu. 2014. <span>“The Hsp90-Dependent Proteome Is Conserved and Enriched for Hub Proteins with High Levels of Protein<span></span>Protein Connectivity.”</span> <em>Genome Biology and Evolution</em> 6 (10): 2851–65. <a href="https://doi.org/10.1093/gbe/evu226">https://doi.org/10.1093/gbe/evu226</a>.
</div>
<div id="ref-Hohrman2021" class="csl-entry" role="doc-biblioentry">
Hohrman, Kaitlyn, Davi Gonçalves, Kevin A Morano, and Jill L Johnson. 2021. <span>“Disrupting Progression of the Yeast Hsp90 Folding Pathway at Different Transition Points Results in Client-Specific Maturation Defects.”</span> Edited by A Mitchell. <em>Genetics</em> 217 (3). <a href="https://doi.org/10.1093/genetics/iyab009">https://doi.org/10.1093/genetics/iyab009</a>.
</div>
<div id="ref-Rizzolo2019" class="csl-entry" role="doc-biblioentry">
Rizzolo, Kamran, and Walid A. Houry. 2019. <span>“Multiple Functionalities of Molecular Chaperones Revealed Through Systematic Mapping of Their Interaction Networks.”</span> <em>Journal of Biological Chemistry</em> 294 (6): 2142–50. <a href="https://doi.org/10.1074/jbc.tm118.002805">https://doi.org/10.1074/jbc.tm118.002805</a>.
</div>
<div id="ref-Zhao2005a" class="csl-entry" role="doc-biblioentry">
Zhao, Rongmin, Mike Davey, Ya-Chieh Hsu, Pia Kaplanek, Amy Tong, Ainslie B. Parsons, Nevan Krogan, et al. 2005. <span>“Navigating the Chaperone Network: An Integrative Map of Physical and Genetic Interactions Mediated by the Hsp90 Chaperone.”</span> <em>Cell</em> 120 (5): 715–27. <a href="https://doi.org/10.1016/j.cell.2004.12.024">https://doi.org/10.1016/j.cell.2004.12.024</a>.
</div>
</div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb13" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "BCB 520 - The Final Project"</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "Mutations in Molecular Chaperone: Hsp90 Elicit Different Proteomic Responses"</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="co">    toc: false</span></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="co">    echo: true</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Erick Rios"</span></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "2023-05-03"</span></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a><span class="an">categories:</span><span class="co"> [Portfolio, DataViz, Network, Project, Hsp90, Mutations]</span></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a><span class="an">image:</span><span class="co"> "yeastbud.png"</span></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a><span class="an">description:</span><span class="co"> "Network visualization and analysis of proteomic responses to Hsp90 mutation."</span></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a><span class="an">code-fold:</span><span class="co"> true</span></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a><span class="an">code-tools:</span><span class="co"> true</span></span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a><span class="an">bibliography:</span><span class="co"> references.bib</span></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false</span></span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(igraph)</span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(viridis)</span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a><span class="fu"># The Preamble</span></span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a>The focus of my research is the study of Hsp90 molecular chaperone. Over the decades we come to realize this chaperone sits as a major hub of protein-protein interactions <span class="co">[</span><span class="ot">@Gopinath2014; @Zhao2005a; @Rizzolo2019</span><span class="co">]</span> in which it regulates the folding, activity, and degradation of its clients.</span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-32"><a href="#cb13-32" aria-hidden="true" tabindex="-1"></a>The current study expands the preliminary proteomic analysis previously done in yeast by us<span class="co">[</span><span class="ot">@Hohrman2021</span><span class="co">]</span>, here we use quantitative proteomic analysis to compare extracts from yeast expressing wild-type yeast Hsp90 to nine different mutants. Our results demonstrate that the mutants elicit varied proteomic responses, even though the cells were harvested at temperatures that do not significantly affect growth. Overall, our results suggests:</span>
<span id="cb13-33"><a href="#cb13-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-34"><a href="#cb13-34" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**cellular processes** that may be differentially affected by Hsc82 mutation.</span>
<span id="cb13-35"><a href="#cb13-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-36"><a href="#cb13-36" aria-hidden="true" tabindex="-1"></a><span class="fu"># Data</span></span>
<span id="cb13-37"><a href="#cb13-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-38"><a href="#cb13-38" aria-hidden="true" tabindex="-1"></a><span class="fu">## Significant hits in our Proteomic Data from DIA-MS</span></span>
<span id="cb13-39"><a href="#cb13-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-40"><a href="#cb13-40" aria-hidden="true" tabindex="-1"></a>Data source:</span>
<span id="cb13-41"><a href="#cb13-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-42"><a href="#cb13-42" aria-hidden="true" tabindex="-1"></a><span class="ss">1.  </span>Main data used in this analysis are the results of Data Independent Acquisition Mass Spectrometry (DIA-MS) experiments, a technique for quantitative proteomics. This is the crux of our study. For brevity, the data has been pre-filtered for significant hits of interest for analysis.</span>
<span id="cb13-43"><a href="#cb13-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-44"><a href="#cb13-44" aria-hidden="true" tabindex="-1"></a>A significant hit in our study is a protein whose abundance had a \|Log~2~ Fold Change\| (LogFC) greater than 1.5, relative to wild-type.</span>
<span id="cb13-45"><a href="#cb13-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-48"><a href="#cb13-48" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-49"><a href="#cb13-49" aria-hidden="true" tabindex="-1"></a><span class="co"># Load in our data</span></span>
<span id="cb13-50"><a href="#cb13-50" aria-hidden="true" tabindex="-1"></a>proteomic <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"trimmed_proteomic_Data.csv"</span>)</span>
<span id="cb13-51"><a href="#cb13-51" aria-hidden="true" tabindex="-1"></a><span class="fu">glimpse</span>(proteomic)</span>
<span id="cb13-52"><a href="#cb13-52" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-53"><a href="#cb13-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-54"><a href="#cb13-54" aria-hidden="true" tabindex="-1"></a><span class="fu">### Proteomic Data Dictionary</span></span>
<span id="cb13-55"><a href="#cb13-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-58"><a href="#cb13-58" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-59"><a href="#cb13-59" aria-hidden="true" tabindex="-1"></a>proteomic.dictionary <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"Proteomic_data_dictionary.csv"</span>)</span>
<span id="cb13-60"><a href="#cb13-60" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(proteomic.dictionary)</span>
<span id="cb13-61"><a href="#cb13-61" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-62"><a href="#cb13-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-63"><a href="#cb13-63" aria-hidden="true" tabindex="-1"></a><span class="fu">## List of Hsp90 Interactions</span></span>
<span id="cb13-64"><a href="#cb13-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-65"><a href="#cb13-65" aria-hidden="true" tabindex="-1"></a><span class="ss">2.  </span>Supporting data/information publicly available on *Saccharomyces Genome Database ([SGD](https://yeastgenome.org/ "Saccharomyces Genome Database main page"))*</span>
<span id="cb13-66"><a href="#cb13-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-67"><a href="#cb13-67" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>List of currently known protein **physical** interactors of Hsp90 (Both <span class="co">[</span><span class="ot">Hsc82</span><span class="co">](https://yeastgenome.org/locus/S000004798/interaction "List of Hsc82 genetic and physical interactors")</span> and <span class="co">[</span><span class="ot">Hsp82</span><span class="co">](https://yeastgenome.org/locus/S000006161/interaction "List of Hsp82 genetic and physical interactors")</span> isoforms)</span>
<span id="cb13-68"><a href="#cb13-68" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>List of currently known protein **genetic** interactors of Hsp90 (Both <span class="co">[</span><span class="ot">Hsc82</span><span class="co">](https://yeastgenome.org/locus/S000004798/interaction "List of Hsc82 genetic and physical interactors")</span> and <span class="co">[</span><span class="ot">Hsp82</span><span class="co">](https://yeastgenome.org/locus/S000006161/interaction "List of Hsp82 genetic and physical interactors")</span> isoforms)</span>
<span id="cb13-69"><a href="#cb13-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-72"><a href="#cb13-72" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-73"><a href="#cb13-73" aria-hidden="true" tabindex="-1"></a>Hsp90.Interactors <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"SGD_All_interactions_Hsp90_with_Refs_trimmed.csv"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb13-74"><a href="#cb13-74" aria-hidden="true" tabindex="-1"></a><span class="fu">glimpse</span>(Hsp90.Interactors)</span>
<span id="cb13-75"><a href="#cb13-75" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-76"><a href="#cb13-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-77"><a href="#cb13-77" aria-hidden="true" tabindex="-1"></a><span class="fu">### Hsp90 Interaction Dictionary</span></span>
<span id="cb13-78"><a href="#cb13-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-81"><a href="#cb13-81" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-82"><a href="#cb13-82" aria-hidden="true" tabindex="-1"></a>Hsp90.Interactors.dictionary <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"SGD_Hsp90_interactions_dictionary.csv"</span>)</span>
<span id="cb13-83"><a href="#cb13-83" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(Hsp90.Interactors.dictionary)</span>
<span id="cb13-84"><a href="#cb13-84" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-85"><a href="#cb13-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-86"><a href="#cb13-86" aria-hidden="true" tabindex="-1"></a><span class="fu">## GO Slim Mapper results</span></span>
<span id="cb13-87"><a href="#cb13-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-88"><a href="#cb13-88" aria-hidden="true" tabindex="-1"></a>Hits from our data analyzed by <span class="co">[</span><span class="ot">Gene Ontology (GO) Slim Mapper</span><span class="co">](https://yeastgenome.org/goSlimMapper "Gene Ontology tool to classify and group genes/proteins")</span></span>
<span id="cb13-89"><a href="#cb13-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-92"><a href="#cb13-92" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-93"><a href="#cb13-93" aria-hidden="true" tabindex="-1"></a>GO.terms <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="at">file =</span> <span class="st">"GO Slim Mapper Results.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb13-94"><a href="#cb13-94" aria-hidden="true" tabindex="-1"></a><span class="fu">glimpse</span>(GO.terms)</span>
<span id="cb13-95"><a href="#cb13-95" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-96"><a href="#cb13-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-97"><a href="#cb13-97" aria-hidden="true" tabindex="-1"></a><span class="fu">### GO Slim Mapper results Dictionary</span></span>
<span id="cb13-98"><a href="#cb13-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-101"><a href="#cb13-101" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-102"><a href="#cb13-102" aria-hidden="true" tabindex="-1"></a>Go.dictionary <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"GO_process_results_dictionary.csv"</span>)</span>
<span id="cb13-103"><a href="#cb13-103" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(proteomic.dictionary)</span>
<span id="cb13-104"><a href="#cb13-104" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-105"><a href="#cb13-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-106"><a href="#cb13-106" aria-hidden="true" tabindex="-1"></a><span class="fu"># Visualizations</span></span>
<span id="cb13-107"><a href="#cb13-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-108"><a href="#cb13-108" aria-hidden="true" tabindex="-1"></a><span class="fu">## Figure 1</span></span>
<span id="cb13-109"><a href="#cb13-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-112"><a href="#cb13-112" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-113"><a href="#cb13-113" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb13-114"><a href="#cb13-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-115"><a href="#cb13-115" aria-hidden="true" tabindex="-1"></a><span class="do">### Trim and format the Gene Ontology: Process data</span></span>
<span id="cb13-116"><a href="#cb13-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-117"><a href="#cb13-117" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate GO process enrichment</span></span>
<span id="cb13-118"><a href="#cb13-118" aria-hidden="true" tabindex="-1"></a>GO.terms<span class="sc">$</span>CLUSTER_FREQUENCY <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">sub</span>(<span class="st">"%"</span>, <span class="st">""</span>, GO.terms<span class="sc">$</span>CLUSTER_FREQUENCY))<span class="sc">/</span><span class="dv">100</span></span>
<span id="cb13-119"><a href="#cb13-119" aria-hidden="true" tabindex="-1"></a>GO.terms<span class="sc">$</span>GENOME_FREQUENCY <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">sub</span>(<span class="st">"%"</span>, <span class="st">""</span>, GO.terms<span class="sc">$</span>GENOME_FREQUENCY))<span class="sc">/</span><span class="dv">100</span></span>
<span id="cb13-120"><a href="#cb13-120" aria-hidden="true" tabindex="-1"></a>GO.terms<span class="sc">$</span>Enrichment <span class="ot">&lt;-</span> GO.terms<span class="sc">$</span>CLUSTER_FREQUENCY<span class="sc">/</span>GO.terms<span class="sc">$</span>GENOME_FREQUENCY</span>
<span id="cb13-121"><a href="#cb13-121" aria-hidden="true" tabindex="-1"></a><span class="co"># Take top 5 enriched processes that have at least 5 proteins annotated to a cellular process.</span></span>
<span id="cb13-122"><a href="#cb13-122" aria-hidden="true" tabindex="-1"></a>GO.terms5 <span class="ot">&lt;-</span> GO.terms[GO.terms<span class="sc">$</span>NUM_LIST_ANNOTATIONS <span class="sc">&gt;=</span> <span class="dv">5</span>,]</span>
<span id="cb13-123"><a href="#cb13-123" aria-hidden="true" tabindex="-1"></a>GO.terms5 <span class="ot">&lt;-</span> GO.terms5[<span class="fu">order</span>(GO.terms5<span class="sc">$</span>Enrichment, <span class="at">decreasing =</span> T),]</span>
<span id="cb13-124"><a href="#cb13-124" aria-hidden="true" tabindex="-1"></a>GO.terms5 <span class="ot">&lt;-</span> GO.terms5[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">9</span>,<span class="dv">10</span>)]</span>
<span id="cb13-125"><a href="#cb13-125" aria-hidden="true" tabindex="-1"></a><span class="co"># Format for network</span></span>
<span id="cb13-126"><a href="#cb13-126" aria-hidden="true" tabindex="-1"></a>GO.terms5.split <span class="ot">&lt;-</span> <span class="fu">separate_rows</span>(GO.terms5, ANNOTATED_GENES, <span class="at">sep =</span> <span class="st">", "</span>)</span>
<span id="cb13-127"><a href="#cb13-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-128"><a href="#cb13-128" aria-hidden="true" tabindex="-1"></a><span class="do">###</span></span>
<span id="cb13-129"><a href="#cb13-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-130"><a href="#cb13-130" aria-hidden="true" tabindex="-1"></a><span class="co"># Make initial network</span></span>
<span id="cb13-131"><a href="#cb13-131" aria-hidden="true" tabindex="-1"></a>vertices <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(GO.terms5.split<span class="sc">$</span>TERM, GO.terms5.split<span class="sc">$</span>ANNOTATED_GENES))</span>
<span id="cb13-132"><a href="#cb13-132" aria-hidden="true" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">graph_from_data_frame</span>(GO.terms5.split[,<span class="dv">2</span><span class="sc">:</span><span class="dv">3</span>], <span class="at">directed =</span> F, <span class="at">vertices =</span> vertices)</span>
<span id="cb13-133"><a href="#cb13-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-134"><a href="#cb13-134" aria-hidden="true" tabindex="-1"></a><span class="do">###</span></span>
<span id="cb13-135"><a href="#cb13-135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-136"><a href="#cb13-136" aria-hidden="true" tabindex="-1"></a><span class="co"># create layout using force directed algorithm</span></span>
<span id="cb13-137"><a href="#cb13-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-138"><a href="#cb13-138" aria-hidden="true" tabindex="-1"></a>layout <span class="ot">&lt;-</span> <span class="fu">layout_with_fr</span>(g, <span class="at">weights =</span> GO.terms5.split<span class="sc">$</span>Enrichment, <span class="at">coords =</span> <span class="fu">matrix</span>(<span class="at">data =</span> <span class="dv">0</span>, <span class="at">nrow =</span> <span class="dv">91</span>, <span class="at">ncol =</span> <span class="dv">2</span>))</span>
<span id="cb13-139"><a href="#cb13-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-140"><a href="#cb13-140" aria-hidden="true" tabindex="-1"></a><span class="do">###</span></span>
<span id="cb13-141"><a href="#cb13-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-142"><a href="#cb13-142" aria-hidden="true" tabindex="-1"></a><span class="co"># Add mutant nodes </span></span>
<span id="cb13-143"><a href="#cb13-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-144"><a href="#cb13-144" aria-hidden="true" tabindex="-1"></a>mutants <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">c</span>(<span class="fu">cos</span>(<span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">2</span><span class="sc">*</span>pi, <span class="at">length.out =</span> <span class="dv">10</span>)), </span>
<span id="cb13-145"><a href="#cb13-145" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">sin</span>(<span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">2</span><span class="sc">*</span>pi, <span class="at">length.out =</span> <span class="dv">10</span>))), <span class="at">ncol =</span> <span class="dv">2</span>)<span class="sc">*</span><span class="dv">11</span> <span class="co"># Create coordinates for a ring.</span></span>
<span id="cb13-146"><a href="#cb13-146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-147"><a href="#cb13-147" aria-hidden="true" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">add_vertices</span>(g, <span class="dv">9</span>)</span>
<span id="cb13-148"><a href="#cb13-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-149"><a href="#cb13-149" aria-hidden="true" tabindex="-1"></a>mutant.names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"R46G"</span>, <span class="st">"G309S"</span>, <span class="st">"S481Y"</span>, <span class="st">"A583T"</span>, <span class="st">"K102E"</span>, <span class="st">"S25P"</span>, <span class="st">"Q380K"</span>, <span class="st">"G424D"</span>, <span class="st">"W296A"</span>)</span>
<span id="cb13-150"><a href="#cb13-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-151"><a href="#cb13-151" aria-hidden="true" tabindex="-1"></a><span class="fu">V</span>(g)<span class="sc">$</span>name[<span class="dv">92</span><span class="sc">:</span><span class="dv">100</span>] <span class="ot">&lt;-</span> mutant.names</span>
<span id="cb13-152"><a href="#cb13-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-153"><a href="#cb13-153" aria-hidden="true" tabindex="-1"></a><span class="co"># and edges between mutants and proteins/genes</span></span>
<span id="cb13-154"><a href="#cb13-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-155"><a href="#cb13-155" aria-hidden="true" tabindex="-1"></a>edgelist <span class="ot">&lt;-</span> proteomic[,<span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">5</span>)][proteomic<span class="sc">$</span>Gene <span class="sc">%in%</span> GO.terms5.split<span class="sc">$</span>ANNOTATED_GENES,]</span>
<span id="cb13-156"><a href="#cb13-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-157"><a href="#cb13-157" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert edge list to indices</span></span>
<span id="cb13-158"><a href="#cb13-158" aria-hidden="true" tabindex="-1"></a>edgelist<span class="sc">$</span>from <span class="ot">&lt;-</span> <span class="fu">match</span>(edgelist<span class="sc">$</span>contrast, <span class="fu">V</span>(g)<span class="sc">$</span>name)</span>
<span id="cb13-159"><a href="#cb13-159" aria-hidden="true" tabindex="-1"></a>edgelist<span class="sc">$</span>to <span class="ot">&lt;-</span> <span class="fu">match</span>(edgelist<span class="sc">$</span>Gene, <span class="fu">V</span>(g)<span class="sc">$</span>name)</span>
<span id="cb13-160"><a href="#cb13-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-161"><a href="#cb13-161" aria-hidden="true" tabindex="-1"></a><span class="co"># create a list of alternating values from 'from' and 'to' columns</span></span>
<span id="cb13-162"><a href="#cb13-162" aria-hidden="true" tabindex="-1"></a>edges <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rbind</span>(edgelist<span class="sc">$</span>from, edgelist<span class="sc">$</span>to))</span>
<span id="cb13-163"><a href="#cb13-163" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-164"><a href="#cb13-164" aria-hidden="true" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">add_edges</span>(g, <span class="at">edges =</span> edges)</span>
<span id="cb13-165"><a href="#cb13-165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-166"><a href="#cb13-166" aria-hidden="true" tabindex="-1"></a><span class="do">###</span></span>
<span id="cb13-167"><a href="#cb13-167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-168"><a href="#cb13-168" aria-hidden="true" tabindex="-1"></a><span class="co"># Now add the nodes for proteins that were either not in the top 5 enriched processes or could not be annotated to a GO process at all.</span></span>
<span id="cb13-169"><a href="#cb13-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-170"><a href="#cb13-170" aria-hidden="true" tabindex="-1"></a><span class="do">###</span></span>
<span id="cb13-171"><a href="#cb13-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-172"><a href="#cb13-172" aria-hidden="true" tabindex="-1"></a><span class="co"># Assign node colors based on physical, genetic, both (mixed), interaction with Hsp90 or neither (unknown)</span></span>
<span id="cb13-173"><a href="#cb13-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-174"><a href="#cb13-174" aria-hidden="true" tabindex="-1"></a>proteomic<span class="sc">$</span>Interaction <span class="ot">&lt;-</span> <span class="st">"Unknown"</span></span>
<span id="cb13-175"><a href="#cb13-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-176"><a href="#cb13-176" aria-hidden="true" tabindex="-1"></a>proteomic<span class="sc">$</span>Interaction[proteomic<span class="sc">$</span>Systemic.name <span class="sc">%in%</span> Hsp90.Interactors<span class="sc">$</span>Systemic.Name[Hsp90.Interactors<span class="sc">$</span>TYPE<span class="sc">==</span><span class="st">"Genetic"</span>]] <span class="ot">&lt;-</span> <span class="st">"Genetic"</span></span>
<span id="cb13-177"><a href="#cb13-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-178"><a href="#cb13-178" aria-hidden="true" tabindex="-1"></a>proteomic<span class="sc">$</span>Interaction[proteomic<span class="sc">$</span>Systemic.name <span class="sc">%in%</span> Hsp90.Interactors<span class="sc">$</span>Systemic.Name[Hsp90.Interactors<span class="sc">$</span>TYPE<span class="sc">==</span><span class="st">"Physical"</span>]] <span class="ot">&lt;-</span> <span class="st">"Physical"</span></span>
<span id="cb13-179"><a href="#cb13-179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-180"><a href="#cb13-180" aria-hidden="true" tabindex="-1"></a>proteomic<span class="sc">$</span>Interaction[proteomic<span class="sc">$</span>Systemic.name <span class="sc">%in%</span> Hsp90.Interactors<span class="sc">$</span>Systemic.Name[Hsp90.Interactors<span class="sc">$</span>TYPE<span class="sc">==</span><span class="st">"Physical"</span>] <span class="sc">&amp;</span></span>
<span id="cb13-181"><a href="#cb13-181" aria-hidden="true" tabindex="-1"></a>                        proteomic<span class="sc">$</span>Systemic.name <span class="sc">%in%</span> Hsp90.Interactors<span class="sc">$</span>Systemic.Name[Hsp90.Interactors<span class="sc">$</span>TYPE<span class="sc">==</span><span class="st">"Genetic"</span>]] <span class="ot">&lt;-</span> <span class="st">"Mixed"</span></span>
<span id="cb13-182"><a href="#cb13-182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-183"><a href="#cb13-183" aria-hidden="true" tabindex="-1"></a><span class="co"># Define color palette for the color blind plebs, just kidding, but not really</span></span>
<span id="cb13-184"><a href="#cb13-184" aria-hidden="true" tabindex="-1"></a>colors <span class="ot">&lt;-</span> <span class="fu">viridis</span>(<span class="at">n =</span> <span class="dv">4</span>, <span class="at">option =</span> <span class="st">"D"</span>, <span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">begin =</span> <span class="dv">0</span>, <span class="at">end =</span> <span class="dv">1</span>, <span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>)</span>
<span id="cb13-185"><a href="#cb13-185" aria-hidden="true" tabindex="-1"></a>colors.process <span class="ot">&lt;-</span> <span class="fu">viridis</span>(<span class="at">n =</span> <span class="dv">5</span>, <span class="at">option =</span> <span class="st">"F"</span>, <span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">begin =</span> <span class="dv">0</span>, <span class="at">end =</span> <span class="dv">1</span>, <span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>)</span>
<span id="cb13-186"><a href="#cb13-186" aria-hidden="true" tabindex="-1"></a>proteomic<span class="sc">$</span>Int.color <span class="ot">&lt;-</span> colors[<span class="fu">match</span>(proteomic<span class="sc">$</span>Interaction, <span class="fu">unique</span>(proteomic<span class="sc">$</span>Interaction))]</span>
<span id="cb13-187"><a href="#cb13-187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-188"><a href="#cb13-188" aria-hidden="true" tabindex="-1"></a><span class="fu">V</span>(g)<span class="sc">$</span>color <span class="ot">&lt;-</span> <span class="fu">c</span>(colors.process, </span>
<span id="cb13-189"><a href="#cb13-189" aria-hidden="true" tabindex="-1"></a>                proteomic<span class="sc">$</span>Int.color[<span class="fu">match</span>(<span class="fu">V</span>(g)<span class="sc">$</span>name, proteomic<span class="sc">$</span>Gene)][<span class="dv">6</span><span class="sc">:</span><span class="dv">91</span>],</span>
<span id="cb13-190"><a href="#cb13-190" aria-hidden="true" tabindex="-1"></a>                <span class="fu">rep</span>(<span class="st">"white"</span>, <span class="dv">9</span>)</span>
<span id="cb13-191"><a href="#cb13-191" aria-hidden="true" tabindex="-1"></a>                )</span>
<span id="cb13-192"><a href="#cb13-192" aria-hidden="true" tabindex="-1"></a><span class="fu">V</span>(g)<span class="sc">$</span>color[<span class="dv">23</span>] <span class="ot">&lt;-</span> <span class="st">"#35B779FF"</span></span>
<span id="cb13-193"><a href="#cb13-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-194"><a href="#cb13-194" aria-hidden="true" tabindex="-1"></a><span class="do">###</span></span>
<span id="cb13-195"><a href="#cb13-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-196"><a href="#cb13-196" aria-hidden="true" tabindex="-1"></a><span class="co"># Modify node size, shape, and labels to display</span></span>
<span id="cb13-197"><a href="#cb13-197" aria-hidden="true" tabindex="-1"></a>term.count <span class="ot">&lt;-</span>  <span class="fu">length</span>(<span class="fu">unique</span>(GO.terms5.split<span class="sc">$</span>TERM))</span>
<span id="cb13-198"><a href="#cb13-198" aria-hidden="true" tabindex="-1"></a>node.count <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">unique</span>(GO.terms5.split<span class="sc">$</span>ANNOTATED_GENES))</span>
<span id="cb13-199"><a href="#cb13-199" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-200"><a href="#cb13-200" aria-hidden="true" tabindex="-1"></a><span class="fu">V</span>(g)<span class="sc">$</span>size <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="dv">12</span>, term.count), <span class="fu">rep</span>(<span class="dv">5</span>, node.count), <span class="fu">rep</span>(<span class="dv">0</span>, <span class="dv">9</span>))</span>
<span id="cb13-201"><a href="#cb13-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-202"><a href="#cb13-202" aria-hidden="true" tabindex="-1"></a><span class="fu">V</span>(g)<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">""</span>, <span class="dv">91</span>), mutant.names)</span>
<span id="cb13-203"><a href="#cb13-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-204"><a href="#cb13-204" aria-hidden="true" tabindex="-1"></a><span class="fu">V</span>(g)<span class="sc">$</span>shape <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"circle"</span>, term.count), <span class="fu">rep</span>(<span class="st">"square"</span> , node.count), <span class="fu">rep</span>(<span class="st">"none"</span>, <span class="dv">9</span>))</span>
<span id="cb13-205"><a href="#cb13-205" aria-hidden="true" tabindex="-1"></a><span class="do">###</span></span>
<span id="cb13-206"><a href="#cb13-206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-207"><a href="#cb13-207" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(g,</span>
<span id="cb13-208"><a href="#cb13-208" aria-hidden="true" tabindex="-1"></a>     <span class="at">layout =</span> <span class="fu">rbind</span>(layout, mutants[<span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]),</span>
<span id="cb13-209"><a href="#cb13-209" aria-hidden="true" tabindex="-1"></a>     <span class="at">margin =</span> <span class="sc">-</span>.<span class="dv">8</span>, <span class="co"># allow the network to use up more of the margin</span></span>
<span id="cb13-210"><a href="#cb13-210" aria-hidden="true" tabindex="-1"></a>     <span class="at">vertex.label.cex =</span> <span class="dv">1</span>,</span>
<span id="cb13-211"><a href="#cb13-211" aria-hidden="true" tabindex="-1"></a>     <span class="at">vertex.label.font =</span> <span class="dv">2</span>, <span class="co"># Bold</span></span>
<span id="cb13-212"><a href="#cb13-212" aria-hidden="true" tabindex="-1"></a>     <span class="at">vertex.label.color=</span> <span class="st">"black"</span>,</span>
<span id="cb13-213"><a href="#cb13-213" aria-hidden="true" tabindex="-1"></a>     <span class="at">edge_arrow_mode =</span> <span class="dv">0</span>,</span>
<span id="cb13-214"><a href="#cb13-214" aria-hidden="true" tabindex="-1"></a>     <span class="at">vertex.color =</span> <span class="fu">V</span>(g)<span class="sc">$</span>color,</span>
<span id="cb13-215"><a href="#cb13-215" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="fl">1.75</span>), <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>, <span class="dv">2</span>) <span class="co"># Increase the plot size to help accommodate the network</span></span>
<span id="cb13-216"><a href="#cb13-216" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb13-217"><a href="#cb13-217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-218"><a href="#cb13-218" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"topleft"</span>, <span class="at">legend=</span><span class="fu">unique</span>(proteomic<span class="sc">$</span>Interaction), <span class="at">fill=</span><span class="fu">unique</span>(<span class="fu">V</span>(g)<span class="sc">$</span>color)[<span class="dv">6</span><span class="sc">:</span><span class="dv">91</span>], <span class="at">title =</span> <span class="st">"Hsp90 Interaction Type"</span>, <span class="at">cex =</span> .<span class="dv">75</span>)</span>
<span id="cb13-219"><a href="#cb13-219" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-220"><a href="#cb13-220" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"bottomleft"</span>, <span class="at">legend=</span> <span class="fu">V</span>(g)<span class="sc">$</span>name[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>], <span class="at">title =</span> <span class="st">"GO Process"</span>, <span class="at">cex =</span> .<span class="dv">75</span>, <span class="at">pch =</span> <span class="dv">21</span>, <span class="at">pt.cex =</span> <span class="dv">2</span>, <span class="at">pt.bg =</span> <span class="fu">V</span>(g)<span class="sc">$</span>color[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>], <span class="at">pt.lwd =</span> <span class="fl">0.5</span>)</span>
<span id="cb13-221"><a href="#cb13-221" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-222"><a href="#cb13-222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-223"><a href="#cb13-223" aria-hidden="true" tabindex="-1"></a><span class="fu">### NODES</span></span>
<span id="cb13-224"><a href="#cb13-224" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-225"><a href="#cb13-225" aria-hidden="true" tabindex="-1"></a>Here we have 3 classes or groups of nodes:</span>
<span id="cb13-226"><a href="#cb13-226" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-227"><a href="#cb13-227" aria-hidden="true" tabindex="-1"></a><span class="ss">1.  </span>Our 9 different Hsc82 mutations</span>
<span id="cb13-228"><a href="#cb13-228" aria-hidden="true" tabindex="-1"></a><span class="ss">2.  </span>The top 5 most enriched cellular processes given by Gene Ontology Slim Mapper tool</span>
<span id="cb13-229"><a href="#cb13-229" aria-hidden="true" tabindex="-1"></a><span class="ss">3.  </span>86 different proteins/genes that were annotated to one or more of the top cellular processes *and* protein abundance was significantly affected in one or more Hsc82 mutants</span>
<span id="cb13-230"><a href="#cb13-230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-231"><a href="#cb13-231" aria-hidden="true" tabindex="-1"></a>Node labels were removed for network clarity, with exception of mutant nodes. The GO process terms were moved to a legend and color was used to identify the GO terms in the figure. The focus of the figure was to identify potential processes, not individual proteins, that may be affected by one or more Hsc82 mutant, therefore labels for individual proteins were removed.</span>
<span id="cb13-232"><a href="#cb13-232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-233"><a href="#cb13-233" aria-hidden="true" tabindex="-1"></a>Node colors were picked in consideration for color blind individuals using the "Viridis" color package.</span>
<span id="cb13-234"><a href="#cb13-234" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-235"><a href="#cb13-235" aria-hidden="true" tabindex="-1"></a>In an effort to help reduce cognitive demand, the node shapes and sizes were adjusted to help visually separate the node classes and direct the viewer from the outer layer of the network, towards the center with increasing detail. Node shape for the mutations were removed and labels were left as the main focus of our study revolves around these mutations. The GO terms were designated as circles and size is larger compared to the individual proteins which were designated as squares. Overall, the increase in shape size moving outward from the center gives a sense of hierarchy of importance.</span>
<span id="cb13-236"><a href="#cb13-236" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-237"><a href="#cb13-237" aria-hidden="true" tabindex="-1"></a>By focusing on the top 5 enriched GO processes, the classic hairball problem is avoided and we can show with visual clarity the cellular processes that would most likely be considered for follow-up experiments or studies.</span>
<span id="cb13-238"><a href="#cb13-238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-239"><a href="#cb13-239" aria-hidden="true" tabindex="-1"></a><span class="fu">### EDGES</span></span>
<span id="cb13-240"><a href="#cb13-240" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-241"><a href="#cb13-241" aria-hidden="true" tabindex="-1"></a>Genes/proteins annotated to given GO terms by GO Slim mapper had edges drawn between the protein and the corresponding GO term(s). A few proteins, were annotated to more than one cellular process.</span>
<span id="cb13-242"><a href="#cb13-242" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-243"><a href="#cb13-243" aria-hidden="true" tabindex="-1"></a>Edges were drawn between proteins and Hsc82 mutations if the protein's abundance was significantly affected by an Hsc82 mutation. Some proteins were affected by more than one mutation and therefore have multiple edges.</span>
<span id="cb13-244"><a href="#cb13-244" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-245"><a href="#cb13-245" aria-hidden="true" tabindex="-1"></a><span class="fu">### LAYOUT</span></span>
<span id="cb13-246"><a href="#cb13-246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-247"><a href="#cb13-247" aria-hidden="true" tabindex="-1"></a>An initial network was made consisting of GO terms and proteins, after which the Fruchterman-Reingold force directed algorithm was employed and the GO term enrichments were used as edge weights.</span>
<span id="cb13-248"><a href="#cb13-248" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-249"><a href="#cb13-249" aria-hidden="true" tabindex="-1"></a>Afterwards, the Hsc82 mutant nodes were manually added around the perimeter of the initial network to avoid mutant nodes from influencing the force directed graph and avoid a network hairball.</span>
<span id="cb13-250"><a href="#cb13-250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-251"><a href="#cb13-251" aria-hidden="true" tabindex="-1"></a><span class="fu"># Conclusions</span></span>
<span id="cb13-252"><a href="#cb13-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-255"><a href="#cb13-255" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-256"><a href="#cb13-256" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(g,</span>
<span id="cb13-257"><a href="#cb13-257" aria-hidden="true" tabindex="-1"></a>     <span class="at">layout =</span> <span class="fu">rbind</span>(layout, mutants[<span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]),</span>
<span id="cb13-258"><a href="#cb13-258" aria-hidden="true" tabindex="-1"></a>     <span class="at">margin =</span> <span class="sc">-</span>.<span class="dv">8</span>, <span class="co"># allow the network to use up more of the margin</span></span>
<span id="cb13-259"><a href="#cb13-259" aria-hidden="true" tabindex="-1"></a>     <span class="at">vertex.label.cex =</span> <span class="dv">1</span>,</span>
<span id="cb13-260"><a href="#cb13-260" aria-hidden="true" tabindex="-1"></a>     <span class="at">vertex.label.font =</span> <span class="dv">2</span>, <span class="co"># Bold</span></span>
<span id="cb13-261"><a href="#cb13-261" aria-hidden="true" tabindex="-1"></a>     <span class="at">vertex.label.color=</span> <span class="st">"black"</span>,</span>
<span id="cb13-262"><a href="#cb13-262" aria-hidden="true" tabindex="-1"></a>     <span class="at">edge_arrow_mode =</span> <span class="dv">0</span>,</span>
<span id="cb13-263"><a href="#cb13-263" aria-hidden="true" tabindex="-1"></a>     <span class="at">vertex.color =</span> <span class="fu">V</span>(g)<span class="sc">$</span>color,</span>
<span id="cb13-264"><a href="#cb13-264" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="fl">1.75</span>), <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>, <span class="dv">2</span>) <span class="co"># Increase the plot size to help accommodate the network</span></span>
<span id="cb13-265"><a href="#cb13-265" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb13-266"><a href="#cb13-266" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-267"><a href="#cb13-267" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"topleft"</span>, <span class="at">legend=</span><span class="fu">unique</span>(proteomic<span class="sc">$</span>Interaction), <span class="at">fill=</span><span class="fu">unique</span>(<span class="fu">V</span>(g)<span class="sc">$</span>color)[<span class="dv">6</span><span class="sc">:</span><span class="dv">91</span>], <span class="at">title =</span> <span class="st">"Hsp90 Interaction Type"</span>, <span class="at">cex =</span> .<span class="dv">75</span>)</span>
<span id="cb13-268"><a href="#cb13-268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-269"><a href="#cb13-269" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"bottomleft"</span>, <span class="at">legend=</span> <span class="fu">V</span>(g)<span class="sc">$</span>name[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>], <span class="at">title =</span> <span class="st">"GO Process"</span>, <span class="at">cex =</span> .<span class="dv">75</span>, <span class="at">pch =</span> <span class="dv">21</span>, <span class="at">pt.cex =</span> <span class="dv">2</span>, <span class="at">pt.bg =</span> <span class="fu">V</span>(g)<span class="sc">$</span>color[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>], <span class="at">pt.lwd =</span> <span class="fl">0.5</span>)</span>
<span id="cb13-270"><a href="#cb13-270" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-271"><a href="#cb13-271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-272"><a href="#cb13-272" aria-hidden="true" tabindex="-1"></a>Preliminary network analysis reveals</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>